// @sanity/ui/dist/index.mjs
import { X as getTheme_v2 } from "./_chunks/theme.mjs";
import { $ as responsiveInputPaddingStyle, A as useMediaIndex, At as parseColor, B as _cardColorStyle, C as BoundaryElementContext, Ct as isHTMLTextAreaElement, D as Grid, Dt as hexToRgb, E as Inline, Et as createColorTheme, F as _ResizeObserver, Ft as studioTheme, G as SpanWithTextOverflow, H as Flex, I as useCustomValidity, J as textInputBaseStyle, K as responsiveRadiusStyle, L as useClickOutsideEvent, M as useGlobalKeyDown, Mt as rgbToHsl, N as useElementSize, Nt as rgba, O as usePrefersReducedMotion, Ot as hslToRgb, P as _elementSizeObserver, Pt as screen, Q as responsiveInputPaddingIconRightStyle, R as Card, S as isRecord, St as isHTMLSelectElement, T as KBD, Tt as EMPTY_RECORD, U as Text, V as Spinner, W as Box, X as textInputRepresentationStyle, Y as textInputFontSizeStyle, Z as textInputRootStyle, _ as usePortal, _t as containsOrEqualsElement, a as MenuDivider, at as useRootTheme, b as LayerProvider, bt as isHTMLElement, c as Tooltip, ct as responsiveTextFont, d as useDelayedState, dt as _fillCSSObject, et as focusRingBorderStyle, f as TextInput, ft as _getArrayProp, g as Portal, gt as _isEnterToClickElement, h as PortalProvider, ht as rem, i as MenuGroup, it as ThemeColorProvider, j as useMatchMedia, jt as rgbToHex, k as usePrefersDark, kt as multiply, l as useTooltipDelayGroup, lt as responsiveTextAlignStyle, m as Popover, mt as _responsive, n as Tab, nt as useTheme, o as Menu, ot as ThemeProvider, p as Stack, pt as _getResponsiveSpace, q as responsivePaddingStyle, r as MenuItem, rt as useTheme_v2, s as Hotkeys, st as createGlobalScopedContext, t as TabList, tt as focusRingStyle, u as TooltipDelayGroupContext, ut as responsiveFont, v as Layer, vt as isHTMLAnchorElement, w as Arrow, wt as EMPTY_ARRAY, x as useBoundaryElement, xt as isHTMLInputElement, y as useLayer, yt as isHTMLButtonElement, z as Button } from "./_chunks/tabList.mjs";
import { c } from "react-compiler-runtime";
import { CheckmarkIcon, ChevronDownIcon, CloseIcon, RemoveIcon, SpinnerIcon, ToggleArrowRightIcon } from "@sanity/icons";
import { Children, Component, Fragment, Suspense, cloneElement, forwardRef, isValidElement, lazy, startTransition, useContext, useEffect, useId, useImperativeHandle, useMemo, useReducer, useRef, useState, useSyncExternalStore } from "react";
import ReactIs from "react-is";
import { css, keyframes, styled } from "styled-components";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";
import { AnimatePresence, motion } from "motion/react";
/**
* @internal
*/
function _raf(fn) {
	let frameId = requestAnimationFrame(fn);
	return () => {
		cancelAnimationFrame(frameId);
	};
}
/**
* @internal
*/
function _raf2(fn) {
	let innerDispose = null, outerDispose = _raf(() => {
		innerDispose = _raf(fn);
	});
	return () => {
		innerDispose && innerDispose(), outerDispose();
	};
}
/**
* @internal
*/
function _hasFocus(element) {
	return !!document.activeElement && element.contains(document.activeElement);
}
/**
* @internal
*/
function isFocusable(element) {
	return element.tabIndex > 0 || element.tabIndex === 0 && element.getAttribute("tabIndex") !== null ? !0 : isHTMLAnchorElement(element) ? !!element.href && element.rel !== "ignore" : isHTMLInputElement(element) ? element.type !== "hidden" && element.type !== "file" && !element.disabled : isHTMLButtonElement(element) || isHTMLSelectElement(element) || isHTMLTextAreaElement(element) ? !element.disabled : !1;
}
/**
* @internal
*/
function attemptFocus(element) {
	if (!isFocusable(element)) return !1;
	try {
		element.focus();
	} catch {}
	return document.activeElement === element;
}
/**
* @internal
*/
function focusFirstDescendant(element) {
	for (let i = 0; i < element.childNodes.length; i++) {
		let child = element.childNodes[i];
		if (isHTMLElement(child) && (attemptFocus(child) || focusFirstDescendant(child))) return !0;
	}
	return !1;
}
/**
* @internal
*/
function focusLastDescendant(element) {
	for (let i = element.childNodes.length - 1; i >= 0; i--) {
		let child = element.childNodes[i];
		if (isHTMLElement(child) && (attemptFocus(child) || focusLastDescendant(child))) return !0;
	}
	return !1;
}
/**
* @internal
*/
function _isScrollable(el) {
	if (!(el instanceof Element)) return !1;
	let style = window.getComputedStyle(el);
	return style.overflowX.includes("auto") || style.overflowX.includes("scroll") || style.overflowY.includes("auto") || style.overflowY.includes("scroll");
}
/**
* Get responsive CSS for the `code` font style.
* @internal
*/
function responsiveCodeFontStyle(props) {
	return responsiveFont("code", props);
}
/**
* Get responsive CSS for the `heading` font style.
* @internal
*/
function responsiveHeadingFont(props) {
	return responsiveFont("heading", props);
}
/**
* Get responsive CSS for the `label` font style.
* @internal
*/
function responsiveLabelFont(props) {
	return responsiveFont("label", props);
}
function labelBaseStyle(props) {
	let { $accent, $muted } = props, { font } = getTheme_v2(props.theme);
	return css`text-transform:uppercase;${$accent && css`color:var(--card-accent-fg-color);`} ${$muted && css`color:var(--card-muted-fg-color);`} & code{font-family:${font.code.family};border-radius:1px;}& a{text-decoration:none;border-radius:1px;}& svg{display:inline;}& [data-sanity-icon]{vertical-align:baseline;}`;
}
const StyledLabel = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledLabel",
	componentId: "sc-cvgwlt-0"
})(responsiveLabelFont, responsiveTextAlignStyle, labelBaseStyle), LabelComponent = forwardRef(function(props, ref) {
	let $ = c(26), accent, align, childrenProp, restProps, t0, t1, textOverflow, weight;
	$[0] === props ? (accent = $[1], align = $[2], childrenProp = $[3], restProps = $[4], t0 = $[5], t1 = $[6], textOverflow = $[7], weight = $[8]) : ({accent, align, children: childrenProp, muted: t0, size: t1, textOverflow, weight, ...restProps} = props, $[0] = props, $[1] = accent, $[2] = align, $[3] = childrenProp, $[4] = restProps, $[5] = t0, $[6] = t1, $[7] = textOverflow, $[8] = weight);
	let muted = t0 === void 0 ? !1 : t0, size = t1 === void 0 ? 2 : t1, children = childrenProp;
	if (textOverflow === "ellipsis") {
		let t2;
		$[9] === children ? t2 = $[10] : (t2 = /* @__PURE__ */ jsx(SpanWithTextOverflow, { children }), $[9] = children, $[10] = t2), children = t2;
	} else {
		let t2;
		$[11] === children ? t2 = $[12] : (t2 = /* @__PURE__ */ jsx("span", { children }), $[11] = children, $[12] = t2), children = t2;
	}
	let t2;
	$[13] === align ? t2 = $[14] : (t2 = _getArrayProp(align), $[13] = align, $[14] = t2);
	let t3;
	$[15] === size ? t3 = $[16] : (t3 = _getArrayProp(size), $[15] = size, $[16] = t3);
	let t4;
	return $[17] !== accent || $[18] !== children || $[19] !== muted || $[20] !== ref || $[21] !== restProps || $[22] !== t2 || $[23] !== t3 || $[24] !== weight ? (t4 = /* @__PURE__ */ jsx(StyledLabel, {
		"data-ui": "Label",
		...restProps,
		$accent: accent,
		$align: t2,
		$muted: muted,
		$size: t3,
		$weight: weight,
		ref,
		children
	}), $[17] = accent, $[18] = children, $[19] = muted, $[20] = ref, $[21] = restProps, $[22] = t2, $[23] = t3, $[24] = weight, $[25] = t4) : t4 = $[25], t4;
});
LabelComponent.displayName = "ForwardRef(Label)";
/**
* Typographic labels.
*
* @public
*/
const Label = LabelComponent, avatarStyle = {
	root: avatarRootStyle,
	arrow: avatarArrowStyle,
	bgStroke: avatarBgStrokeStyle,
	stroke: avatarStrokeStyle,
	initials: avatarInitialsStyle,
	image: avatarImageStyle
};
function avatarArrowStyle() {
	return {
		position: "absolute",
		boxSizing: "border-box",
		zIndex: "0",
		opacity: "0",
		transition: "all 0.2s linear",
		transform: "rotate(-90deg) translate3d(0, 6px, 0)",
		left: 0,
		right: 0,
		top: 0,
		bottom: 0,
		"& > svg": {
			width: "11px",
			height: "7px",
			position: "absolute",
			top: "-5px",
			left: "50%",
			transform: "translateX(-6px)",
			"&:not([hidden])": { display: "block" }
		},
		"[data-arrow-position='inside'] > &": {
			transform: "rotate(-90deg) translate3d(0, 6px, 0)",
			opacity: "0"
		},
		"[data-arrow-position='top'] > &": {
			opacity: "1",
			transform: "rotate(0deg)"
		},
		"[data-arrow-position='bottom'] > &": {
			opacity: "1",
			transform: "rotate(-180deg)"
		}
	};
}
function avatarRootStyle(props) {
	let { $color } = props, { avatar } = getTheme_v2(props.theme);
	return {
		"--avatar-bg-color": `var(--card-avatar-${$color}-bg-color)`,
		"--avatar-fg-color": `var(--card-avatar-${$color}-fg-color)`,
		backgroundColor: "var(--avatar-bg-color)",
		position: "relative",
		boxSizing: "border-box",
		userSelect: "none",
		boxShadow: "0 0 0 1px var(--card-bg-color)",
		"&[data-status=\"inactive\"]": { opacity: "0.5" },
		"&>svg": { "&:not([hidden])": { display: "block" } },
		"&[data-as=\"button\"]": {
			WebkitFontSmoothing: "inherit",
			appearance: "none",
			margin: 0,
			padding: 0,
			border: 0,
			font: "inherit",
			color: "inherit",
			outline: "none",
			"&:focus": { boxShadow: focusRingStyle({ focusRing: avatar.focusRing }) },
			"&:focus:not(:focus-visible)": { boxShadow: "none" }
		}
	};
}
function responsiveAvatarSizeStyle(props) {
	let { avatar, media } = getTheme_v2(props.theme);
	return _responsive(media, props.$size, (size) => {
		let avatarSize = avatar.sizes[size] || avatar.sizes[0];
		return {
			width: rem(avatarSize.size),
			height: rem(avatarSize.size),
			borderRadius: rem(avatarSize.size / 2),
			"&>svg": {
				width: rem(avatarSize.size),
				height: rem(avatarSize.size),
				borderRadius: rem(avatarSize.size / 2)
			}
		};
	});
}
function avatarImageStyle() {
	return { position: "relative" };
}
function avatarInitialsStyle() {
	return {
		width: "100%",
		height: "100%",
		color: "var(--avatar-fg-color)",
		alignItems: "center",
		justifyContent: "center",
		textTransform: "uppercase",
		textAlign: "center",
		borderRadius: "50%",
		"&:not([hidden])": { display: "flex" }
	};
}
function avatarBgStrokeStyle() {
	return {
		strokeWidth: "4px",
		stroke: "var(--card-bg-color)"
	};
}
function avatarStrokeStyle() {
	return {
		strokeWidth: "2px",
		stroke: "var(--avatar-bg-color)",
		"[data-status=\"editing\"] &": {
			strokeDasharray: "2 4",
			strokeLinecap: "round"
		}
	};
}
const StyledAvatar = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledAvatar",
	componentId: "sc-jxq2u8-0"
})(responsiveAvatarSizeStyle, avatarStyle.root), Arrow$1 = /* @__PURE__ */ styled.div.withConfig({
	displayName: "Arrow",
	componentId: "sc-jxq2u8-1"
})(avatarStyle.arrow), BgStroke = /* @__PURE__ */ styled.ellipse.withConfig({
	displayName: "BgStroke",
	componentId: "sc-jxq2u8-2"
})(avatarStyle.bgStroke), Stroke = /* @__PURE__ */ styled.ellipse.withConfig({
	displayName: "Stroke",
	componentId: "sc-jxq2u8-3"
})(avatarStyle.stroke), Initials = /* @__PURE__ */ styled.div.withConfig({
	displayName: "Initials",
	componentId: "sc-jxq2u8-4"
})(avatarStyle.initials), InitialsLabel = /* @__PURE__ */ styled(Label).withConfig({
	displayName: "InitialsLabel",
	componentId: "sc-jxq2u8-5"
})({ color: "inherit" }), AvatarImage = /* @__PURE__ */ styled.svg.withConfig({
	displayName: "AvatarImage",
	componentId: "sc-jxq2u8-6"
})(avatarStyle.image), AvatarComponent = forwardRef(function(props, ref) {
	let $ = c(46), __unstable_hideInnerStroke, animateArrowFrom, arrowPositionProp, asProp, initials, onImageLoadError, restProps, src, t0, t1, t2, title;
	$[0] === props ? (__unstable_hideInnerStroke = $[1], animateArrowFrom = $[2], arrowPositionProp = $[3], asProp = $[4], initials = $[5], onImageLoadError = $[6], restProps = $[7], src = $[8], t0 = $[9], t1 = $[10], t2 = $[11], title = $[12]) : ({__unstable_hideInnerStroke, as: asProp, color: t0, src, title, initials, onImageLoadError, arrowPosition: arrowPositionProp, animateArrowFrom, status: t1, size: t2, ...restProps} = props, $[0] = props, $[1] = __unstable_hideInnerStroke, $[2] = animateArrowFrom, $[3] = arrowPositionProp, $[4] = asProp, $[5] = initials, $[6] = onImageLoadError, $[7] = restProps, $[8] = src, $[9] = t0, $[10] = t1, $[11] = t2, $[12] = title);
	let color = t0 === void 0 ? "gray" : t0, status = t1 === void 0 ? "online" : t1, sizeProp = t2 === void 0 ? 1 : t2, { avatar } = useTheme_v2(), as = ReactIs.isValidElementType(asProp) ? asProp : "div", size = _getArrayProp(sizeProp), _sizeRem = (avatar.sizes[size[0]] || avatar.sizes[0]).size, _radius = _sizeRem / 2, elementId = useId(), [arrowPosition, setArrowPosition] = useState(animateArrowFrom || arrowPositionProp || "inside"), [imageFailed, setImageFailed] = useState(!1), imageId = `avatar-image-${elementId}`, t3, t4;
	$[13] !== arrowPosition || $[14] !== arrowPositionProp ? (t3 = () => {
		if (arrowPosition === arrowPositionProp) return;
		let raf = requestAnimationFrame(() => setArrowPosition(arrowPositionProp));
		return () => cancelAnimationFrame(raf);
	}, t4 = [arrowPosition, arrowPositionProp], $[13] = arrowPosition, $[14] = arrowPositionProp, $[15] = t3, $[16] = t4) : (t3 = $[15], t4 = $[16]), useEffect(t3, t4);
	let t5, t6;
	$[17] === src ? (t5 = $[18], t6 = $[19]) : (t5 = () => {
		src && setImageFailed(!1);
	}, t6 = [src], $[17] = src, $[18] = t5, $[19] = t6), useEffect(t5, t6);
	let t7;
	$[20] === onImageLoadError ? t7 = $[21] : (t7 = () => {
		setImageFailed(!0), onImageLoadError && onImageLoadError(/* @__PURE__ */ Error("Avatar: the image failed to load"));
	}, $[20] = onImageLoadError, $[21] = t7);
	let handleImageError = t7, T0 = StyledAvatar, t8 = typeof as == "string" ? as : void 0, t10;
	$[22] === color ? t10 = $[23] : (t10 = /* @__PURE__ */ jsx(Arrow$1, { children: /* @__PURE__ */ jsx("svg", {
		width: "11",
		height: "7",
		viewBox: "0 0 11 7",
		fill: "none",
		children: /* @__PURE__ */ jsx("path", {
			d: "M6.67948 1.50115L11 7L0 7L4.32052 1.50115C4.92109 0.736796 6.07891 0.736795 6.67948 1.50115Z",
			fill: color
		})
	}) }), $[22] = color, $[23] = t10);
	let t11;
	$[24] !== __unstable_hideInnerStroke || $[25] !== _radius || $[26] !== _sizeRem || $[27] !== handleImageError || $[28] !== imageFailed || $[29] !== imageId || $[30] !== src ? (t11 = !imageFailed && src && /* @__PURE__ */ jsxs(AvatarImage, {
		viewBox: `0 0 ${_sizeRem} ${_sizeRem}`,
		fill: "none",
		children: [
			/* @__PURE__ */ jsx("defs", { children: /* @__PURE__ */ jsx("pattern", {
				id: imageId,
				patternContentUnits: "objectBoundingBox",
				width: "1",
				height: "1",
				children: /* @__PURE__ */ jsx("image", {
					href: src,
					width: "1",
					height: "1",
					onError: handleImageError
				})
			}) }),
			/* @__PURE__ */ jsx("circle", {
				cx: _radius,
				cy: _radius,
				r: _radius,
				fill: `url(#${imageId})`
			}),
			!__unstable_hideInnerStroke && /* @__PURE__ */ jsx(BgStroke, {
				cx: _radius,
				cy: _radius,
				rx: _radius,
				ry: _radius,
				vectorEffect: "non-scaling-stroke"
			}),
			/* @__PURE__ */ jsx(Stroke, {
				cx: _radius,
				cy: _radius,
				rx: _radius,
				ry: _radius,
				vectorEffect: "non-scaling-stroke"
			})
		]
	}), $[24] = __unstable_hideInnerStroke, $[25] = _radius, $[26] = _sizeRem, $[27] = handleImageError, $[28] = imageFailed, $[29] = imageId, $[30] = src, $[31] = t11) : t11 = $[31];
	let t12 = (imageFailed || !src) && initials && /* @__PURE__ */ jsx(Fragment$1, { children: /* @__PURE__ */ jsx(Initials, { children: /* @__PURE__ */ jsx(InitialsLabel, {
		forwardedAs: "span",
		size: size.map(_temp$4),
		weight: "medium",
		children: initials
	}) }) }), t13;
	return $[32] !== T0 || $[33] !== arrowPosition || $[34] !== as || $[35] !== color || $[36] !== ref || $[37] !== restProps || $[38] !== size || $[39] !== status || $[40] !== t10 || $[41] !== t11 || $[42] !== t12 || $[43] !== t8 || $[44] !== title ? (t13 = /* @__PURE__ */ jsxs(T0, {
		as,
		"data-as": t8,
		"data-ui": "Avatar",
		...restProps,
		$color: color,
		$size: size,
		"aria-label": title,
		"data-arrow-position": arrowPosition,
		"data-status": status,
		ref,
		title,
		children: [
			t10,
			t11,
			t12
		]
	}), $[32] = T0, $[33] = arrowPosition, $[34] = as, $[35] = color, $[36] = ref, $[37] = restProps, $[38] = size, $[39] = status, $[40] = t10, $[41] = t11, $[42] = t12, $[43] = t8, $[44] = title, $[45] = t13) : t13 = $[45], t13;
});
AvatarComponent.displayName = "ForwardRef(Avatar)";
/**
* Avatars are used to represent people and other agents (e.g. bots).
*
* @public
*/
const Avatar = AvatarComponent;
function _temp$4(s) {
	return s === 1 ? 1 : s === 2 ? 3 : s === 3 ? 5 : 0;
}
function _responsiveAvatarCounterSizeStyle(props) {
	let { avatar, media } = getTheme_v2(props.theme);
	return _responsive(media, props.$size, (size) => {
		let avatarSize = avatar.sizes[size];
		return avatarSize ? {
			borderRadius: rem(avatarSize.size / 2),
			minWidth: rem(avatarSize.size),
			height: rem(avatarSize.size)
		} : EMPTY_RECORD;
	});
}
function _avatarCounterBaseStyle(props) {
	let { space } = getTheme_v2(props.theme);
	return css`align-items:center;justify-content:center;box-sizing:border-box;user-select:none;color:inherit;color:var(--card-fg-color);background:var(--card-bg-color);box-shadow:0 0 0 1px var(--card-bg-color),inset 0 0 0 1px var(--card-hairline-hard-color);padding:0 ${rem(space[2])};&:not([hidden]){display:flex;}`;
}
const StyledAvatarCounter = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledAvatarCounter",
	componentId: "sc-s1txqz-0"
})(_responsiveAvatarCounterSizeStyle, _avatarCounterBaseStyle), AvatarCounter = forwardRef(function(props, ref) {
	let $ = c(20), { count, size: t0 } = props, sizeProp = t0 === void 0 ? 1 : t0, T0, T1, t1, t2, t3, t4, t5;
	if ($[0] !== ref || $[1] !== sizeProp) {
		let size = _getArrayProp(sizeProp);
		T1 = StyledAvatarCounter, t3 = size, t4 = "AvatarCounter", t5 = ref, T0 = Label, t1 = "span", t2 = size.map(_temp$3), $[0] = ref, $[1] = sizeProp, $[2] = T0, $[3] = T1, $[4] = t1, $[5] = t2, $[6] = t3, $[7] = t4, $[8] = t5;
	} else T0 = $[2], T1 = $[3], t1 = $[4], t2 = $[5], t3 = $[6], t4 = $[7], t5 = $[8];
	let t6;
	$[9] !== T0 || $[10] !== count || $[11] !== t1 || $[12] !== t2 ? (t6 = /* @__PURE__ */ jsx(T0, {
		as: t1,
		size: t2,
		weight: "medium",
		children: count
	}), $[9] = T0, $[10] = count, $[11] = t1, $[12] = t2, $[13] = t6) : t6 = $[13];
	let t7;
	return $[14] !== T1 || $[15] !== t3 || $[16] !== t4 || $[17] !== t5 || $[18] !== t6 ? (t7 = /* @__PURE__ */ jsx(T1, {
		$size: t3,
		"data-ui": t4,
		ref: t5,
		children: t6
	}), $[14] = T1, $[15] = t3, $[16] = t4, $[17] = t5, $[18] = t6, $[19] = t7) : t7 = $[19], t7;
});
AvatarCounter.displayName = "ForwardRef(AvatarCounter)";
function _temp$3(s) {
	return s === 1 ? 1 : s === 2 ? 3 : s === 3 ? 5 : 0;
}
const BASE_STYLES = css`white-space:nowrap;& > div{vertical-align:top;&:not([hidden]){display:inline-block;}}`;
function avatarStackStyle() {
	return BASE_STYLES;
}
function responsiveAvatarStackSizeStyle(props) {
	let { avatar, media } = getTheme_v2(props.theme);
	return _responsive(media, props.$size, (size) => {
		let avatarSize = avatar.sizes[size];
		return avatarSize ? { "& > div + div": { marginLeft: rem(avatarSize.distance) } } : EMPTY_RECORD;
	});
}
const StyledAvatarStack = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledAvatarStack",
	componentId: "sc-ks59o1-0"
})(responsiveAvatarStackSizeStyle, avatarStackStyle), AvatarStack = forwardRef(function(props, ref) {
	let $ = c(38), childrenProp, restProps, t0, t1;
	$[0] === props ? (childrenProp = $[1], restProps = $[2], t0 = $[3], t1 = $[4]) : ({children: childrenProp, maxLength: t0, size: t1, ...restProps} = props, $[0] = props, $[1] = childrenProp, $[2] = restProps, $[3] = t0, $[4] = t1);
	let maxLengthProp = t0 === void 0 ? 4 : t0, sizeProp = t1 === void 0 ? 1 : t1, T0, t2, t3, t4, t5, t6, t7, t8;
	if ($[5] !== childrenProp || $[6] !== maxLengthProp || $[7] !== ref || $[8] !== restProps || $[9] !== sizeProp) {
		let children = Children.toArray(childrenProp).filter(isValidElement), maxLength = Math.max(maxLengthProp, 0), t9;
		$[18] === sizeProp ? t9 = $[19] : (t9 = _getArrayProp(sizeProp), $[18] = sizeProp, $[19] = t9);
		let size = t9, len = children.length, extraCount = len - (maxLength - 1), visibleChildren = extraCount > 1 ? children.slice(extraCount, len) : children;
		T0 = StyledAvatarStack, t2 = "AvatarStack", t3 = restProps, t4 = ref, t5 = size, $[20] !== len || $[21] !== size ? (t6 = len === 0 && /* @__PURE__ */ jsx("div", { children: /* @__PURE__ */ jsx(AvatarCounter, {
			count: len,
			size
		}) }), $[20] = len, $[21] = size, $[22] = t6) : t6 = $[22], $[23] !== extraCount || $[24] !== len || $[25] !== size ? (t7 = len !== 0 && extraCount > 1 && /* @__PURE__ */ jsx("div", { children: /* @__PURE__ */ jsx(AvatarCounter, {
			count: extraCount,
			size
		}) }), $[23] = extraCount, $[24] = len, $[25] = size, $[26] = t7) : t7 = $[26];
		let t10;
		$[27] === size ? t10 = $[28] : (t10 = (child, childIndex) => /* @__PURE__ */ jsx("div", { children: cloneElement(child, { size }) }, String(childIndex)), $[27] = size, $[28] = t10), t8 = visibleChildren.map(t10), $[5] = childrenProp, $[6] = maxLengthProp, $[7] = ref, $[8] = restProps, $[9] = sizeProp, $[10] = T0, $[11] = t2, $[12] = t3, $[13] = t4, $[14] = t5, $[15] = t6, $[16] = t7, $[17] = t8;
	} else T0 = $[10], t2 = $[11], t3 = $[12], t4 = $[13], t5 = $[14], t6 = $[15], t7 = $[16], t8 = $[17];
	let t9;
	return $[29] !== T0 || $[30] !== t2 || $[31] !== t3 || $[32] !== t4 || $[33] !== t5 || $[34] !== t6 || $[35] !== t7 || $[36] !== t8 ? (t9 = /* @__PURE__ */ jsxs(T0, {
		"data-ui": t2,
		...t3,
		ref: t4,
		$size: t5,
		children: [
			t6,
			t7,
			t8
		]
	}), $[29] = T0, $[30] = t2, $[31] = t3, $[32] = t4, $[33] = t5, $[34] = t6, $[35] = t7, $[36] = t8, $[37] = t9) : t9 = $[37], t9;
});
AvatarStack.displayName = "ForwardRef(AvatarStack)";
function badgeStyle(props) {
	let { $tone } = props;
	return {
		"--card-bg-color": `var(--card-badge-${$tone}-bg-color)`,
		"--card-fg-color": `var(--card-badge-${$tone}-fg-color)`,
		backgroundColor: "var(--card-bg-color)",
		cursor: "default",
		"&:not([hidden])": {
			display: "inline-block",
			verticalAlign: "top"
		}
	};
}
const StyledBadge = /* @__PURE__ */ styled(Box).withConfig({
	displayName: "StyledBadge",
	componentId: "sc-p8fi74-0"
})(responsiveRadiusStyle, badgeStyle), BadgeComponent = forwardRef(function(props, ref) {
	let $ = c(21), children, restProps, t0, t1, t2, t3;
	if ($[0] !== props) {
		let { children: t4, fontSize: t5, mode: _deprecated_mode, padding: t6, radius: t7, tone: t8, ...t9 } = props;
		children = t4, t0 = t5, t1 = t6, t2 = t7, t3 = t8, restProps = t9, $[0] = props, $[1] = children, $[2] = restProps, $[3] = t0, $[4] = t1, $[5] = t2, $[6] = t3;
	} else children = $[1], restProps = $[2], t0 = $[3], t1 = $[4], t2 = $[5], t3 = $[6];
	let fontSize = t0 === void 0 ? 1 : t0, padding = t1 === void 0 ? 1 : t1, radius = t2 === void 0 ? "full" : t2, tone = t3 === void 0 ? "default" : t3, t4;
	$[7] === radius ? t4 = $[8] : (t4 = _getArrayProp(radius), $[7] = radius, $[8] = t4);
	let t5;
	$[9] === padding ? t5 = $[10] : (t5 = _getArrayProp(padding), $[9] = padding, $[10] = t5);
	let t6;
	$[11] !== children || $[12] !== fontSize ? (t6 = /* @__PURE__ */ jsx(Text, {
		size: fontSize,
		children
	}), $[11] = children, $[12] = fontSize, $[13] = t6) : t6 = $[13];
	let t7;
	return $[14] !== ref || $[15] !== restProps || $[16] !== t4 || $[17] !== t5 || $[18] !== t6 || $[19] !== tone ? (t7 = /* @__PURE__ */ jsx(StyledBadge, {
		"data-ui": "Badge",
		...restProps,
		$tone: tone,
		$radius: t4,
		padding: t5,
		ref,
		children: t6
	}), $[14] = ref, $[15] = restProps, $[16] = t4, $[17] = t5, $[18] = t6, $[19] = tone, $[20] = t7) : t7 = $[20], t7;
});
BadgeComponent.displayName = "ForwardRef(Badge)";
/**
* Badges are used to tag resources.
*
* @public
*/
const Badge = BadgeComponent;
/**
* @deprecated instead of `useArrayProp(width)` use `Array.isArray(width) ? width : [width]` instead
* @beta
*/
function useArrayProp(val, defaultVal) {
	return useMemo(() => _getArrayProp(val, defaultVal), [val, defaultVal]);
}
function _getElements(element, elementsArg) {
	let ret = [element];
	for (let el of elementsArg) Array.isArray(el) ? ret.push(...el) : ret.push(el);
	return ret.filter(Boolean);
}
/**
* @public
* @deprecated replaced by the new `useClickOutsideEvent` hook, instead of:
* ```tsx
* const [button, setButtonElement] = useState(null)
* useClickOutside((event) => {}, [button])
* return <button ref={setButtonElement} />
* ```
* do:
* ```tsx
* const buttonRef = useRef()
* useClickOutsideEvent((event) => {}, () => [buttonRef.current])
* return <button ref={buttonRef} />
* ```
*/
function useClickOutside(listener, elementsArg = EMPTY_ARRAY, boundaryElement) {
	let [element, setElement] = useState(null), [elements, setElements] = useState(() => _getElements(element, elementsArg)), elementsRef = useRef(elements);
	return useEffect(() => {
		let prevElements = elementsRef.current, nextElements = _getElements(element, elementsArg);
		if (prevElements.length !== nextElements.length) {
			setElements(nextElements), elementsRef.current = nextElements;
			return;
		}
		for (let el of prevElements) if (!nextElements.includes(el)) {
			setElements(nextElements), elementsRef.current = nextElements;
			return;
		}
		for (let el of nextElements) if (!prevElements.includes(el)) {
			setElements(nextElements), elementsRef.current = nextElements;
			return;
		}
	}, [element, elementsArg]), useEffect(() => {
		if (!listener) return;
		let handleWindowMouseDown = (evt) => {
			let target = evt.target;
			if (target instanceof Node && !(boundaryElement && !boundaryElement.contains(target))) {
				for (let el of elements) if (target === el || el.contains(target)) return;
				listener(evt);
			}
		};
		return window.addEventListener("mousedown", handleWindowMouseDown), () => {
			window.removeEventListener("mousedown", handleWindowMouseDown);
		};
	}, [
		boundaryElement,
		listener,
		elements
	]), setElement;
}
/**
* Subscribe to the rect of a DOM element.
* @beta
*
* @deprecated Use `useElementSize` instead
*/
function useElementRect(element) {
	return useElementSize(element)?._contentRect || null;
}
/**
* @beta
* @deprecated use `useImperativeHandle` instead
* @example
* ```diff
* -const ref = useForwardedRef(forwardedRef)
* +const ref = useRef(null)
* +useImperativeHandle(forwardedRef, () => ref.current)
* ```
*/
function useForwardedRef(ref) {
	let innerRef = useRef(null);
	return useImperativeHandle(ref, () => innerRef.current), innerRef;
}
function checkboxBaseStyles() {
	return css`position:relative;display:inline-block;`;
}
function inputElementStyles(props) {
	let { color, input, radius } = getTheme_v2(props.theme), { focusRing } = input.checkbox;
	return css`position:absolute;top:0;left:0;width:100%;height:100%;outline:none;opacity:0;z-index:1;padding:0;margin:0;& + span{position:relative;display:block;height:${rem(input.checkbox.size)};width:${rem(input.checkbox.size)};box-sizing:border-box;box-shadow:${focusRingBorderStyle({
		color: color.input.default.enabled.border,
		width: input.border.width
	})};border-radius:${rem(radius[2])};line-height:1;background-color:${color.input.default.enabled.bg};& > svg{display:block;position:absolute;opacity:0;height:100%;width:100%;& > path{vector-effect:non-scaling-stroke;stroke-width:1.5px !important;}}}&:checked + span{background:${color.input.default.enabled.fg};box-shadow:${focusRingBorderStyle({
		color: color.input.default.enabled.fg,
		width: input.border.width
	})};color:${color.input.default.enabled.bg};}&:not(:disabled):focus:focus-visible + span{box-shadow:${focusRingStyle({ focusRing })};}&:not(:disabled):focus:focus-visible&:checked + span{box-shadow:${focusRingStyle({ focusRing: {
		width: 1,
		offset: 1
	} })};}&[data-error] + span{background-color:${color.input.invalid.enabled.border};box-shadow:${focusRingBorderStyle({
		width: input.border.width,
		color: color.input.invalid.enabled.muted.bg
	})};color:${color.input.default.disabled.fg};}&[data-error]&:checked + span{background-color:${color.input.invalid.enabled.muted.bg};color:${color.input.default.enabled.bg};}&[data-error]&:checked&:not(:disabled):focus:focus-visible + span{box-shadow:${focusRingStyle({
		border: {
			width: input.border.width,
			color: color.input.invalid.readOnly.muted.bg
		},
		focusRing: {
			width: 1,
			offset: 1
		}
	})};}&:disabled + span{background-color:${color.input.default.disabled.bg};box-shadow:${focusRingBorderStyle({
		width: input.border.width,
		color: color.input.default.disabled.border
	})};color:${color.input.default.disabled.fg};}&:disabled&:checked + span{background-color:${color.input.default.disabled.muted.bg};}&[data-read-only] + span{background-color:${color.input.default.readOnly.bg};box-shadow:${focusRingBorderStyle({
		width: input.border.width,
		color: color.input.default.readOnly.border
	})};color:${color.input.default.readOnly.fg};}&[data-read-only]&:checked + span{background-color:${color.input.default.readOnly.muted.bg};}&:checked + span > svg:first-child{opacity:1;}&:indeterminate + span > svg:last-child{opacity:1;}`;
}
const StyledCheckbox = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledCheckbox",
	componentId: "sc-mje2ol-0"
})(checkboxBaseStyles), Input$4 = /* @__PURE__ */ styled.input.withConfig({
	displayName: "Input",
	componentId: "sc-mje2ol-1"
})(inputElementStyles), Checkbox = forwardRef(function(props, forwardedRef) {
	let $ = c(25), checked, className, customValidity, disabled, indeterminate, readOnly, restProps, style;
	$[0] === props ? (checked = $[1], className = $[2], customValidity = $[3], disabled = $[4], indeterminate = $[5], readOnly = $[6], restProps = $[7], style = $[8]) : ({checked, className, disabled, indeterminate, customValidity, readOnly, style, ...restProps} = props, $[0] = props, $[1] = checked, $[2] = className, $[3] = customValidity, $[4] = disabled, $[5] = indeterminate, $[6] = readOnly, $[7] = restProps, $[8] = style);
	let ref = useRef(null), t0;
	$[9] === Symbol.for("react.memo_cache_sentinel") ? (t0 = () => ref.current, $[9] = t0) : t0 = $[9], useImperativeHandle(forwardedRef, t0);
	let t1, t2;
	$[10] === indeterminate ? (t1 = $[11], t2 = $[12]) : (t1 = () => {
		ref.current && (ref.current.indeterminate = indeterminate || !1);
	}, t2 = [indeterminate], $[10] = indeterminate, $[11] = t1, $[12] = t2), useEffect(t1, t2), useCustomValidity(ref, customValidity);
	let t3 = !disabled && readOnly ? "" : void 0, t4 = customValidity ? "" : void 0, t5 = disabled || readOnly, t6;
	$[13] !== checked || $[14] !== readOnly || $[15] !== restProps || $[16] !== t3 || $[17] !== t4 || $[18] !== t5 ? (t6 = /* @__PURE__ */ jsx(Input$4, {
		"data-read-only": t3,
		"data-error": t4,
		...restProps,
		checked,
		disabled: t5,
		type: "checkbox",
		readOnly,
		ref
	}), $[13] = checked, $[14] = readOnly, $[15] = restProps, $[16] = t3, $[17] = t4, $[18] = t5, $[19] = t6) : t6 = $[19];
	let t7;
	$[20] === Symbol.for("react.memo_cache_sentinel") ? (t7 = /* @__PURE__ */ jsxs("span", { children: [/* @__PURE__ */ jsx(CheckmarkIcon, {}), /* @__PURE__ */ jsx(RemoveIcon, {})] }), $[20] = t7) : t7 = $[20];
	let t8;
	return $[21] !== className || $[22] !== style || $[23] !== t6 ? (t8 = /* @__PURE__ */ jsxs(StyledCheckbox, {
		className,
		"data-ui": "Checkbox",
		style,
		children: [t6, t7]
	}), $[21] = className, $[22] = style, $[23] = t6, $[24] = t8) : t8 = $[24], t8;
});
Checkbox.displayName = "ForwardRef(Checkbox)";
function codeSyntaxHighlightingStyle({ theme }) {
	let { color: { syntax: color } } = getTheme_v2(theme);
	return {
		"&.atrule": { color: color.atrule },
		"&.attr-name": { color: color.attrName },
		"&.attr-value": { color: color.attrValue },
		"&.attribute": { color: color.attribute },
		"&.boolean": { color: color.boolean },
		"&.builtin": { color: color.builtin },
		"&.cdata": { color: color.cdata },
		"&.char": { color: color.char },
		"&.class": { color: color.class },
		"&.class-name": { color: color.className },
		"&.comment": { color: color.comment },
		"&.constant": { color: color.constant },
		"&.deleted": { color: color.deleted },
		"&.doctype": { color: color.doctype },
		"&.entity": { color: color.entity },
		"&.function": { color: color.function },
		"&.hexcode": { color: color.hexcode },
		"&.id": { color: color.id },
		"&.important": { color: color.important },
		"&.inserted": { color: color.inserted },
		"&.keyword": { color: color.keyword },
		"&.number": { color: color.number },
		"&.operator": { color: color.operator },
		"&.prolog": { color: color.prolog },
		"&.property": { color: color.property },
		"&.pseudo-class": { color: color.pseudoClass },
		"&.pseudo-element": { color: color.pseudoElement },
		"&.punctuation": { color: color.punctuation },
		"&.regex": { color: color.regex },
		"&.selector": { color: color.selector },
		"&.string": { color: color.string },
		"&.symbol": { color: color.symbol },
		"&.tag": { color: color.tag },
		"&.unit": { color: color.unit },
		"&.url": { color: color.url },
		"&.variable": { color: color.variable }
	};
}
function codeBaseStyle() {
	return css`color:var(--card-code-fg-color);& code{font-family:inherit;&.refractor .token{${codeSyntaxHighlightingStyle}}}& a{color:inherit;text-decoration:underline;border-radius:1px;}& svg{display:inline;}& [data-sanity-icon]{vertical-align:baseline;}`;
}
const LazyRefractor = lazy(() => import("./_chunks/refractor.mjs")), StyledCode = /* @__PURE__ */ styled.pre.withConfig({
	displayName: "StyledCode",
	componentId: "sc-botxhl-0"
})(codeBaseStyle, responsiveCodeFontStyle), CodeComponent = forwardRef(function(props, ref) {
	let $ = c(22), children, language, restProps, t0, weight;
	$[0] === props ? (children = $[1], language = $[2], restProps = $[3], t0 = $[4], weight = $[5]) : ({children, language, size: t0, weight, ...restProps} = props, $[0] = props, $[1] = children, $[2] = language, $[3] = restProps, $[4] = t0, $[5] = weight);
	let size = t0 === void 0 ? 2 : t0, t1;
	$[6] === size ? t1 = $[7] : (t1 = _getArrayProp(size), $[6] = size, $[7] = t1);
	let t2 = weight, t3;
	$[8] === children ? t3 = $[9] : (t3 = /* @__PURE__ */ jsx("code", { children }), $[8] = children, $[9] = t3);
	let t4;
	$[10] !== children || $[11] !== language ? (t4 = /* @__PURE__ */ jsx(LazyRefractor, {
		language,
		value: children
	}), $[10] = children, $[11] = language, $[12] = t4) : t4 = $[12];
	let t5;
	$[13] !== t3 || $[14] !== t4 ? (t5 = /* @__PURE__ */ jsx(Suspense, {
		fallback: t3,
		children: t4
	}), $[13] = t3, $[14] = t4, $[15] = t5) : t5 = $[15];
	let t6;
	return $[16] !== ref || $[17] !== restProps || $[18] !== t1 || $[19] !== t2 || $[20] !== t5 ? (t6 = /* @__PURE__ */ jsx(StyledCode, {
		"data-ui": "Code",
		...restProps,
		$size: t1,
		$weight: t2,
		ref,
		children: t5
	}), $[16] = ref, $[17] = restProps, $[18] = t1, $[19] = t2, $[20] = t5, $[21] = t6) : t6 = $[21], t6;
});
CodeComponent.displayName = "ForwardRef(Code)";
/**
* @public
*/
const Code = CodeComponent, BASE_STYLE = {
	width: "100%",
	margin: "0 auto"
};
function containerBaseStyle() {
	return BASE_STYLE;
}
function responsiveContainerWidthStyle(props) {
	let { container, media } = getTheme_v2(props.theme);
	return _responsive(media, props.$width, (val) => ({ maxWidth: val === "auto" ? "none" : rem(container[val]) }));
}
const StyledContainer = /* @__PURE__ */ styled(Box).withConfig({
	displayName: "StyledContainer",
	componentId: "sc-un6r8j-0"
})(containerBaseStyle, responsiveContainerWidthStyle), ContainerComponent = forwardRef(function(props, ref) {
	let $ = c(11), as, restProps, t0;
	$[0] === props ? (as = $[1], restProps = $[2], t0 = $[3]) : ({as, width: t0, ...restProps} = props, $[0] = props, $[1] = as, $[2] = restProps, $[3] = t0);
	let width = t0 === void 0 ? 2 : t0, t1;
	$[4] === width ? t1 = $[5] : (t1 = _getArrayProp(width), $[4] = width, $[5] = t1);
	let t2;
	return $[6] !== as || $[7] !== ref || $[8] !== restProps || $[9] !== t1 ? (t2 = /* @__PURE__ */ jsx(StyledContainer, {
		"data-ui": "Container",
		...restProps,
		$width: t1,
		forwardedAs: as,
		ref
	}), $[6] = as, $[7] = ref, $[8] = restProps, $[9] = t1, $[10] = t2) : t2 = $[10], t2;
});
ContainerComponent.displayName = "ForwardRef(Container)";
/**
* The `Container` component wraps content layout in a defined set of widths.
*
* @public
*/
const Container = ContainerComponent;
function headingBaseStyle(props) {
	let { $accent, $muted } = props, { font } = getTheme_v2(props.theme);
	return css`${$accent && css`color:var(--card-accent-fg-color);`} ${$muted && css`color:var(--card-muted-fg-color);`} & code{font-family:${font.code.family};border-radius:1px;}& a{text-decoration:none;border-radius:1px;color:var(--card-link-color);outline:none;@media (hover:hover){&:hover{text-decoration:underline;}}&:focus{box-shadow:0 0 0 1px var(--card-bg-color),0 0 0 3px var(--card-focus-ring-color);}&:focus:not(:focus-visible){box-shadow:none;}}& strong{font-weight:${font.heading.weights.bold};}& svg{display:inline;}& [data-sanity-icon]{vertical-align:baseline;}`;
}
const StyledHeading = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledHeading",
	componentId: "sc-7dzjaf-0"
})(headingBaseStyle, responsiveTextAlignStyle, responsiveHeadingFont), HeadingComponent = forwardRef(function(props, ref) {
	let $ = c(26), align, childrenProp, restProps, t0, t1, t2, textOverflow, weight;
	$[0] === props ? (align = $[1], childrenProp = $[2], restProps = $[3], t0 = $[4], t1 = $[5], t2 = $[6], textOverflow = $[7], weight = $[8]) : ({accent: t0, align, children: childrenProp, muted: t1, size: t2, textOverflow, weight, ...restProps} = props, $[0] = props, $[1] = align, $[2] = childrenProp, $[3] = restProps, $[4] = t0, $[5] = t1, $[6] = t2, $[7] = textOverflow, $[8] = weight);
	let accent = t0 === void 0 ? !1 : t0, muted = t1 === void 0 ? !1 : t1, size = t2 === void 0 ? 2 : t2, children = childrenProp;
	if (textOverflow === "ellipsis") {
		let t3;
		$[9] === children ? t3 = $[10] : (t3 = /* @__PURE__ */ jsx(SpanWithTextOverflow, { children }), $[9] = children, $[10] = t3), children = t3;
	}
	let t3;
	$[11] === align ? t3 = $[12] : (t3 = _getArrayProp(align), $[11] = align, $[12] = t3);
	let t4;
	$[13] === size ? t4 = $[14] : (t4 = _getArrayProp(size), $[13] = size, $[14] = t4);
	let t5;
	$[15] === children ? t5 = $[16] : (t5 = /* @__PURE__ */ jsx("span", { children }), $[15] = children, $[16] = t5);
	let t6;
	return $[17] !== accent || $[18] !== muted || $[19] !== ref || $[20] !== restProps || $[21] !== t3 || $[22] !== t4 || $[23] !== t5 || $[24] !== weight ? (t6 = /* @__PURE__ */ jsx(StyledHeading, {
		"data-ui": "Heading",
		...restProps,
		$accent: accent,
		$align: t3,
		$muted: muted,
		$size: t4,
		$weight: weight,
		ref,
		children: t5
	}), $[17] = accent, $[18] = muted, $[19] = ref, $[20] = restProps, $[21] = t3, $[22] = t4, $[23] = t5, $[24] = weight, $[25] = t6) : t6 = $[25], t6;
});
HeadingComponent.displayName = "ForwardRef(Heading)";
/**
* Typographic headings.
*
* @public
*/
const Heading = HeadingComponent;
/**
* @public
*/
function BoundaryElementProvider(props) {
	let $ = c(5), { children, element } = props, t0;
	$[0] === element ? t0 = $[1] : (t0 = {
		version: 0,
		element
	}, $[0] = element, $[1] = t0);
	let value = t0, t1;
	return $[2] !== children || $[3] !== value ? (t1 = /* @__PURE__ */ jsx(BoundaryElementContext.Provider, {
		value,
		children
	}), $[2] = children, $[3] = value, $[4] = t1) : t1 = $[4], t1;
}
BoundaryElementProvider.displayName = "BoundaryElementProvider";
/**
* @internal
* @deprecated this component will be removed in the next major release
*/
function ConditionalWrapper({ children, condition, wrapper }) {
	return condition ? wrapper(children) : children;
}
ConditionalWrapper.displayName = "ConditionalWrapper";
function findMaxBreakpoints(media, width) {
	let ret = [];
	for (let i = 0; i < media.length; i += 1) media[i] > width && ret.push(i);
	return ret;
}
function findMinBreakpoints(media, width) {
	let ret = [];
	for (let i = 0; i < media.length; i += 1) media[i] <= width && ret.push(i);
	return ret;
}
/**
* DO NOT USE IN PRODUCTION.
* @beta
*/
const ElementQuery = forwardRef(function(props, forwardedRef) {
	let $ = c(18), theme = useTheme_v2(), _media, children, restProps;
	$[0] === props ? (_media = $[1], children = $[2], restProps = $[3]) : ({children, media: _media, ...restProps} = props, $[0] = props, $[1] = _media, $[2] = children, $[3] = restProps);
	let media = _media ?? theme.media, [element, setElement] = useState(null), width = useElementSize(element)?.border.width ?? window.innerWidth, t0;
	if ($[4] !== media || $[5] !== width) {
		let eq = findMaxBreakpoints(media, width);
		t0 = eq.length ? eq.join(" ") : void 0, $[4] = media, $[5] = width, $[6] = t0;
	} else t0 = $[6];
	let max = t0, t1;
	if ($[7] !== media || $[8] !== width) {
		let eq_0 = findMinBreakpoints(media, width);
		t1 = eq_0.length ? eq_0.join(" ") : void 0, $[7] = media, $[8] = width, $[9] = t1;
	} else t1 = $[9];
	let min = t1, t2, t3;
	$[10] === element ? (t2 = $[11], t3 = $[12]) : (t2 = () => element, t3 = [element], $[10] = element, $[11] = t2, $[12] = t3), useImperativeHandle(forwardedRef, t2, t3);
	let t4;
	return $[13] !== children || $[14] !== max || $[15] !== min || $[16] !== restProps ? (t4 = /* @__PURE__ */ jsx("div", {
		"data-ui": "ElementQuery",
		...restProps,
		"data-eq-max": max,
		"data-eq-min": min,
		ref: setElement,
		children
	}), $[13] = children, $[14] = max, $[15] = min, $[16] = restProps, $[17] = t4) : t4 = $[17], t4;
});
ElementQuery.displayName = "ForwardRef(ElementQuery)";
/**
* DO NOT USE IN PRODUCTION
* @beta
*/
var ErrorBoundary = class extends Component {
	state = { error: null };
	static getDerivedStateFromError(error) {
		return { error };
	}
	componentDidCatch(error, info) {
		this.props.onCatch({
			error,
			info
		});
	}
	render() {
		let { error } = this.state;
		return error ? /* @__PURE__ */ jsx(Code, { children: typeof error?.message == "string" ? error.message : "Error" }) : this.props.children;
	}
};
const StyledSrOnly = styled.div.withConfig({
	displayName: "StyledSrOnly",
	componentId: "sc-wsuz6k-0"
})`display:block;width:0;height:0;position:absolute;overflow:hidden;overflow:clip;`, SrOnly = forwardRef(function(props, ref) {
	let $ = c(4), { as, children } = props, t0;
	return $[0] !== as || $[1] !== children || $[2] !== ref ? (t0 = /* @__PURE__ */ jsx(StyledSrOnly, {
		"aria-hidden": !0,
		as,
		"data-ui": "SrOnly",
		ref,
		children
	}), $[0] = as, $[1] = children, $[2] = ref, $[3] = t0) : t0 = $[3], t0;
});
SrOnly.displayName = "ForwardRef(SrOnly)";
const StyledVirtualList = styled.div.withConfig({
	displayName: "StyledVirtualList",
	componentId: "sc-2o92jb-0"
})`position:relative;`, ItemWrapper = styled.div.withConfig({
	displayName: "ItemWrapper",
	componentId: "sc-2o92jb-1"
})`position:absolute;left:0;right:0;`, VirtualList = forwardRef(function(props, forwardedRef) {
	let $ = c(44), getItemKey, onChange, renderItem, restProps, t0, t1, t2;
	$[0] === props ? (getItemKey = $[1], onChange = $[2], renderItem = $[3], restProps = $[4], t0 = $[5], t1 = $[6], t2 = $[7]) : ({as: t0, gap: t1, getItemKey, items: t2, onChange, renderItem, ...restProps} = props, $[0] = props, $[1] = getItemKey, $[2] = onChange, $[3] = renderItem, $[4] = restProps, $[5] = t0, $[6] = t1, $[7] = t2);
	let as = t0 === void 0 ? "div" : t0, gap = t1 === void 0 ? 0 : t1, t3;
	$[8] === t2 ? t3 = $[9] : (t3 = t2 === void 0 ? [] : t2, $[8] = t2, $[9] = t3);
	let items = t3, { space } = useTheme_v2(), ref = useRef(null), wrapperRef = useRef(null), [scrollTop, setScrollTop] = useState(0), [scrollHeight, setScrollHeight] = useState(0), [itemHeight, setItemHeight] = useState(-1), t4;
	$[10] === Symbol.for("react.memo_cache_sentinel") ? (t4 = () => ref.current, $[10] = t4) : t4 = $[10], useImperativeHandle(forwardedRef, t4);
	let t5;
	$[11] === Symbol.for("react.memo_cache_sentinel") ? (t5 = () => {
		if (!wrapperRef.current) return;
		let firstElement = wrapperRef.current.firstChild;
		firstElement instanceof HTMLElement && setItemHeight(firstElement.offsetHeight);
	}, $[11] = t5) : t5 = $[11];
	let t6;
	$[12] === renderItem ? t6 = $[13] : (t6 = [renderItem], $[12] = renderItem, $[13] = t6), useEffect(t5, t6);
	let t7, t8;
	$[14] === Symbol.for("react.memo_cache_sentinel") ? (t7 = () => {
		if (!ref.current) return;
		let scrollEl = findScrollable(ref.current.parentNode);
		if (scrollEl) {
			if (!(scrollEl instanceof HTMLElement)) return;
			let handleScroll = () => {
				setScrollTop(scrollEl.scrollTop);
			};
			scrollEl.addEventListener("scroll", handleScroll, { passive: !0 });
			let ro = new _ResizeObserver((entries) => {
				setScrollHeight(entries[0].contentRect.height);
			});
			return ro.observe(scrollEl), handleScroll(), () => {
				scrollEl.removeEventListener("scroll", handleScroll), ro.unobserve(scrollEl), ro.disconnect();
			};
		}
		let handleScroll_0 = () => {
			setScrollTop(window.scrollY);
		}, handleResize = () => {
			setScrollHeight(window.innerHeight);
		};
		return window.addEventListener("scroll", handleScroll_0, { passive: !0 }), window.addEventListener("resize", handleResize), setScrollHeight(window.innerHeight), handleScroll_0(), () => {
			window.removeEventListener("scroll", handleScroll_0), window.removeEventListener("resize", handleResize);
		};
	}, t8 = [], $[14] = t7, $[15] = t8) : (t7 = $[14], t8 = $[15]), useEffect(t7, t8);
	let len = items.length, height = itemHeight ? len * (itemHeight + space[gap]) - space[gap] : 0, fromIndex = height ? Math.max(Math.floor(scrollTop / height * len) - 2, 0) : 0, toIndex = height ? Math.ceil((scrollTop + scrollHeight) / height * len) + 1 : 0, t10, t9;
	$[16] !== fromIndex || $[17] !== gap || $[18] !== itemHeight || $[19] !== onChange || $[20] !== scrollHeight || $[21] !== scrollTop || $[22] !== space || $[23] !== toIndex ? (t9 = () => {
		onChange && onChange({
			fromIndex,
			gap: space[gap],
			itemHeight,
			scrollHeight,
			scrollTop,
			toIndex
		});
	}, t10 = [
		fromIndex,
		gap,
		itemHeight,
		onChange,
		scrollHeight,
		scrollTop,
		space,
		toIndex
	], $[16] = fromIndex, $[17] = gap, $[18] = itemHeight, $[19] = onChange, $[20] = scrollHeight, $[21] = scrollTop, $[22] = space, $[23] = toIndex, $[24] = t10, $[25] = t9) : (t10 = $[24], t9 = $[25]), useEffect(t9, t10);
	let t11;
	$[26] !== fromIndex || $[27] !== gap || $[28] !== getItemKey || $[29] !== itemHeight || $[30] !== items || $[31] !== renderItem || $[32] !== space || $[33] !== toIndex ? (t11 = {
		fromIndex,
		gap,
		itemHeight,
		space,
		toIndex,
		getItemKey,
		items,
		renderItem
	}, $[26] = fromIndex, $[27] = gap, $[28] = getItemKey, $[29] = itemHeight, $[30] = items, $[31] = renderItem, $[32] = space, $[33] = toIndex, $[34] = t11) : t11 = $[34];
	let children = useChildren(t11), t12;
	$[35] === height ? t12 = $[36] : (t12 = { height }, $[35] = height, $[36] = t12);
	let t13;
	$[37] !== children || $[38] !== t12 ? (t13 = /* @__PURE__ */ jsx("div", {
		ref: wrapperRef,
		style: t12,
		children
	}), $[37] = children, $[38] = t12, $[39] = t13) : t13 = $[39];
	let t14;
	return $[40] !== as || $[41] !== restProps || $[42] !== t13 ? (t14 = /* @__PURE__ */ jsx(StyledVirtualList, {
		as,
		"data-ui": "VirtualList",
		...restProps,
		ref,
		children: t13
	}), $[40] = as, $[41] = restProps, $[42] = t13, $[43] = t14) : t14 = $[43], t14;
});
VirtualList.displayName = "ForwardRef(VirtualList)";
function useChildren(t0) {
	let $ = c(21), { fromIndex, gap, getItemKey, itemHeight, items, renderItem, space, toIndex } = t0;
	if (!renderItem || items.length === 0) return null;
	if (itemHeight === -1) {
		let t1;
		$[0] !== items[0] || $[1] !== renderItem ? (t1 = renderItem(items[0]), $[0] = items[0], $[1] = renderItem, $[2] = t1) : t1 = $[2];
		let t2;
		return $[3] === t1 ? t2 = $[4] : (t2 = [/* @__PURE__ */ jsx(ItemWrapper, { children: t1 }, 0)], $[3] = t1, $[4] = t2), t2;
	}
	let t1;
	if ($[5] !== fromIndex || $[6] !== gap || $[7] !== getItemKey || $[8] !== itemHeight || $[9] !== items || $[10] !== renderItem || $[11] !== space || $[12] !== toIndex) {
		let t2;
		$[14] !== fromIndex || $[15] !== gap || $[16] !== getItemKey || $[17] !== itemHeight || $[18] !== renderItem || $[19] !== space ? (t2 = (item, _itemIndex) => {
			let itemIndex = fromIndex + _itemIndex, node = renderItem(item), key = getItemKey ? getItemKey(item, itemIndex) : itemIndex;
			return /* @__PURE__ */ jsx(ItemWrapper, {
				style: { top: itemIndex * (itemHeight + space[gap]) },
				children: node
			}, key);
		}, $[14] = fromIndex, $[15] = gap, $[16] = getItemKey, $[17] = itemHeight, $[18] = renderItem, $[19] = space, $[20] = t2) : t2 = $[20], t1 = items.slice(fromIndex, toIndex).map(t2), $[5] = fromIndex, $[6] = gap, $[7] = getItemKey, $[8] = itemHeight, $[9] = items, $[10] = renderItem, $[11] = space, $[12] = toIndex, $[13] = t1;
	} else t1 = $[13];
	return t1;
}
function findScrollable(parentNode) {
	let _scrollEl = parentNode;
	for (; _scrollEl && !_isScrollable(_scrollEl);) _scrollEl = _scrollEl.parentNode;
	return _scrollEl;
}
function radioBaseStyle() {
	return css`position:relative;&:not([hidden]){display:inline-block;}&[data-read-only]{outline:1px solid red;}`;
}
function inputElementStyle(props) {
	let { color, input } = getTheme_v2(props.theme), dist = (input.radio.size - input.radio.markSize) / 2;
	return css`appearance:none;position:absolute;top:0;left:0;opacity:0;height:100%;width:100%;outline:none;z-index:1;padding:0;margin:0;border-radius:${rem(input.radio.size / 2)};border:none;& + span{display:block;position:relative;height:${rem(input.radio.size)};width:${rem(input.radio.size)};border-radius:${rem(input.radio.size / 2)};background:${color.input.default.enabled.bg};box-shadow:${focusRingBorderStyle({
		color: color.input.default.enabled.border,
		width: input.border.width
	})};&::after{content:'';position:absolute;top:${rem(dist)};left:${rem(dist)};height:${rem(input.radio.markSize)};width:${rem(input.radio.markSize)};border-radius:${rem(input.radio.markSize / 2)};background:${color.input.default.enabled.fg};opacity:0;}}&:not(:disabled):focus + span{box-shadow:${focusRingStyle({
		border: {
			width: input.border.width,
			color: color.input.default.enabled.border
		},
		focusRing: input.radio.focusRing
	})};}&:not(:disabled):focus:not(:focus-visible) + span{box-shadow:${focusRingBorderStyle({
		color: color.input.default.enabled.border,
		width: input.border.width
	})};}&:checked + span::after{opacity:1;}&[data-error] + span{background-color:${color.input.invalid.enabled.border};box-shadow:${focusRingBorderStyle({
		width: input.border.width,
		color: color.input.invalid.enabled.muted.bg
	})};&::after{background:${color.input.invalid.enabled.muted.bg};}}&[data-read-only] + span{box-shadow:0 0 0 1px ${color.input.default.readOnly.border};background:${color.input.default.readOnly.bg};&::after{background:${color.input.default.readOnly.border};}}&:not([data-read-only]):disabled + span{box-shadow:0 0 0 1px ${color.input.default.disabled.border};background:${color.input.default.disabled.bg};&::after{background:${color.input.default.disabled.border};}}`;
}
const StyledRadio = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledRadio",
	componentId: "sc-6sng01-0"
})(radioBaseStyle), Input$3 = /* @__PURE__ */ styled.input.withConfig({
	displayName: "Input",
	componentId: "sc-6sng01-1"
})(inputElementStyle), Radio = forwardRef(function(props, forwardedRef) {
	let $ = c(19), className, customValidity, disabled, readOnly, restProps, style;
	$[0] === props ? (className = $[1], customValidity = $[2], disabled = $[3], readOnly = $[4], restProps = $[5], style = $[6]) : ({className, disabled, style, customValidity, readOnly, ...restProps} = props, $[0] = props, $[1] = className, $[2] = customValidity, $[3] = disabled, $[4] = readOnly, $[5] = restProps, $[6] = style);
	let ref = useRef(null), t0;
	$[7] === Symbol.for("react.memo_cache_sentinel") ? (t0 = () => ref.current, $[7] = t0) : t0 = $[7], useImperativeHandle(forwardedRef, t0), useCustomValidity(ref, customValidity);
	let t1 = !disabled && readOnly ? "" : void 0, t2 = customValidity ? "" : void 0, t3 = disabled || readOnly, t4;
	$[8] !== readOnly || $[9] !== restProps || $[10] !== t1 || $[11] !== t2 || $[12] !== t3 ? (t4 = /* @__PURE__ */ jsx(Input$3, {
		"data-read-only": t1,
		"data-error": t2,
		...restProps,
		disabled: t3,
		readOnly,
		ref,
		type: "radio"
	}), $[8] = readOnly, $[9] = restProps, $[10] = t1, $[11] = t2, $[12] = t3, $[13] = t4) : t4 = $[13];
	let t5;
	$[14] === Symbol.for("react.memo_cache_sentinel") ? (t5 = /* @__PURE__ */ jsx("span", {}), $[14] = t5) : t5 = $[14];
	let t6;
	return $[15] !== className || $[16] !== style || $[17] !== t4 ? (t6 = /* @__PURE__ */ jsxs(StyledRadio, {
		className,
		"data-ui": "Radio",
		style,
		children: [t4, t5]
	}), $[15] = className, $[16] = style, $[17] = t4, $[18] = t6) : t6 = $[18], t6;
});
Radio.displayName = "ForwardRef(Radio)";
function rootStyle() {
	return css`position:relative;width:-moz-available;width:-webkit-fill-available;width:stretch;&:not([hidden]){display:inline-block;}`;
}
function inputBaseStyle(props) {
	let { font } = getTheme_v2(props.theme);
	return css`-webkit-font-smoothing:antialiased;appearance:none;border:0;font-family:${font.text.family};color:inherit;width:100%;outline:none;margin:0;&:disabled{opacity:1;}`;
}
function inputColorStyle(props) {
	let { color, input } = getTheme_v2(props.theme);
	return css`background-color:${color.input.default.enabled.bg};color:${color.input.default.enabled.fg};box-shadow:${focusRingBorderStyle({
		color: color.input.default.enabled.border,
		width: input.border.width
	})};@media (hover:hover){&:not(:disabled):hover{background-color:${color.input.default.hovered.bg};color:${color.input.default.hovered.fg};box-shadow:${focusRingBorderStyle({
		color: color.input.default.hovered.border,
		width: input.border.width
	})};}}&:not(:disabled):focus{box-shadow:${focusRingStyle({
		border: {
			width: input.border.width,
			color: color.input.default.enabled.border
		},
		focusRing: input.select.focusRing
	})};}&[data-read-only]{background-color:${color.input.default.readOnly.bg};color:${color.input.default.readOnly.fg};box-shadow:${focusRingBorderStyle({
		color: color.input.default.readOnly.border,
		width: input.border.width
	})};}&:not([data-read-only]):disabled{background-color:${color.input.default.disabled.bg};color:${color.input.default.disabled.fg};box-shadow:${focusRingBorderStyle({
		color: color.input.default.disabled.border,
		width: input.border.width
	})};}`;
}
function textSize(size) {
	return {
		fontSize: rem(size.fontSize),
		lineHeight: `${rem(size.lineHeight)}`
	};
}
function inputTextSizeStyle(props) {
	let { $fontSize } = props, { font, media } = getTheme_v2(props.theme);
	return _responsive(media, $fontSize, (sizeIndex) => textSize(font.text.sizes[sizeIndex] || font.text.sizes[2]));
}
function inputStyle() {
	return [
		responsiveRadiusStyle,
		inputBaseStyle,
		inputColorStyle,
		inputTextSizeStyle,
		responsiveInputPaddingIconRightStyle
	];
}
function iconBoxStyle(props) {
	let { color } = getTheme_v2(props.theme);
	return css`pointer-events:none;position:absolute;top:0;right:0;--card-fg-color:${color.input.default.enabled.fg};@media (hover:hover){select:not(disabled):not(:read-only):hover + &&{--card-fg-color:${color.input.default.hovered.fg};}}select:disabled + &&{--card-fg-color:${color.input.default.disabled.fg};}select[data-read-only] + &&{--card-fg-color:${color.input.default.readOnly.fg};}`;
}
const selectStyle = {
	root: rootStyle,
	input: inputStyle,
	iconBox: iconBoxStyle
}, StyledSelect = /* @__PURE__ */ styled.div.withConfig({
	displayName: "StyledSelect",
	componentId: "sc-a3ltm7-0"
})(selectStyle.root), Input$2 = /* @__PURE__ */ styled.select.withConfig({
	displayName: "Input",
	componentId: "sc-a3ltm7-1"
})(selectStyle.input), IconBox = /* @__PURE__ */ styled(Box).withConfig({
	displayName: "IconBox",
	componentId: "sc-a3ltm7-2"
})(selectStyle.iconBox), Select = forwardRef(function(props, forwardedRef) {
	let $ = c(38), children, customValidity, disabled, gap, readOnly, restProps, t0, t1, t2, t3;
	$[0] === props ? (children = $[1], customValidity = $[2], disabled = $[3], gap = $[4], readOnly = $[5], restProps = $[6], t0 = $[7], t1 = $[8], t2 = $[9], t3 = $[10]) : ({children, customValidity, disabled, fontSize: t0, gap, padding: t1, radius: t2, readOnly, space: t3, ...restProps} = props, $[0] = props, $[1] = children, $[2] = customValidity, $[3] = disabled, $[4] = gap, $[5] = readOnly, $[6] = restProps, $[7] = t0, $[8] = t1, $[9] = t2, $[10] = t3);
	let fontSize = t0 === void 0 ? 2 : t0, padding = t1 === void 0 ? 3 : t1, radius = t2 === void 0 ? 2 : t2, spacing = gap === void 0 ? t3 === void 0 ? 3 : t3 : gap, ref = useRef(null), t4;
	$[11] === Symbol.for("react.memo_cache_sentinel") ? (t4 = () => ref.current, $[11] = t4) : t4 = $[11], useImperativeHandle(forwardedRef, t4), useCustomValidity(ref, customValidity);
	let t5 = !disabled && readOnly ? "" : void 0, t6;
	$[12] === fontSize ? t6 = $[13] : (t6 = _getArrayProp(fontSize), $[12] = fontSize, $[13] = t6);
	let t7;
	$[14] === padding ? t7 = $[15] : (t7 = _getArrayProp(padding), $[14] = padding, $[15] = t7);
	let t8;
	$[16] === radius ? t8 = $[17] : (t8 = _getArrayProp(radius), $[16] = radius, $[17] = t8);
	let t9;
	$[18] === spacing ? t9 = $[19] : (t9 = _getArrayProp(spacing), $[18] = spacing, $[19] = t9);
	let t10 = disabled || readOnly, t11;
	$[20] !== children || $[21] !== restProps || $[22] !== t10 || $[23] !== t5 || $[24] !== t6 || $[25] !== t7 || $[26] !== t8 || $[27] !== t9 ? (t11 = /* @__PURE__ */ jsx(Input$2, {
		"data-read-only": t5,
		"data-ui": "Select",
		...restProps,
		$fontSize: t6,
		$padding: t7,
		$radius: t8,
		$space: t9,
		disabled: t10,
		ref,
		children
	}), $[20] = children, $[21] = restProps, $[22] = t10, $[23] = t5, $[24] = t6, $[25] = t7, $[26] = t8, $[27] = t9, $[28] = t11) : t11 = $[28];
	let t12;
	$[29] === Symbol.for("react.memo_cache_sentinel") ? (t12 = /* @__PURE__ */ jsx(ChevronDownIcon, {}), $[29] = t12) : t12 = $[29];
	let t13;
	$[30] === fontSize ? t13 = $[31] : (t13 = /* @__PURE__ */ jsx(Text, {
		size: fontSize,
		children: t12
	}), $[30] = fontSize, $[31] = t13);
	let t14;
	$[32] !== padding || $[33] !== t13 ? (t14 = /* @__PURE__ */ jsx(IconBox, {
		padding,
		children: t13
	}), $[32] = padding, $[33] = t13, $[34] = t14) : t14 = $[34];
	let t15;
	return $[35] !== t11 || $[36] !== t14 ? (t15 = /* @__PURE__ */ jsxs(StyledSelect, {
		"data-ui": "Select",
		children: [t11, t14]
	}), $[35] = t11, $[36] = t14, $[37] = t15) : t15 = $[37], t15;
});
Select.displayName = "ForwardRef(Select)";
function switchBaseStyles() {
	return css`position:relative;&:not([hidden]){display:inline-block;}`;
}
function switchInputStyles() {
	return css`position:absolute;top:0;right:0;bottom:0;left:0;opacity:0;height:100%;width:100%;outline:none;padding:0;margin:0;z-index:1;`;
}
function switchRepresentationStyles(props) {
	let { color, input } = getTheme_v2(props.theme);
	return css`--switch-bg-color:${color.input.default.enabled.border};--switch-fg-color:${color.input.default.enabled.bg};--switch-box-shadow:none;&:not([hidden]){display:block;}position:relative;width:${rem(input.switch.width)};height:${rem(input.switch.height)};border-radius:${rem(input.switch.height / 2)};pointer-events:none;&:after{content:'';display:block;position:absolute;top:0;left:0;right:0;bottom:0;z-index:1;box-shadow:var(--switch-box-shadow);border-radius:inherit;}input:focus + &&{--switch-box-shadow:${focusRingStyle({ focusRing: input.switch.focusRing })};}input:focus:not(:focus-visible) + &&{--switch-box-shadow:none;}input:checked + &&{--switch-bg-color:${color.input.default.enabled.fg};--switch-fg-color:${color.input.default.enabled.bg};}@media (hover:hover){input:not(:disabled):hover + &&{--switch-bg-color:${color.input.default.hovered.border};--switch-fg-color:${color.input.default.hovered.bg};}input:not(:disabled):checked:hover + &&{--switch-bg-color:${color.input.default.enabled.fg};--switch-fg-color:${color.input.default.enabled.bg};}}input:not([data-read-only]):disabled + &&{--switch-bg-color:${color.input.default.disabled.border};--switch-fg-color:${color.input.default.disabled.bg};}input[data-read-only]:disabled + &&{--switch-bg-color:${color.input.default.readOnly.border};--switch-fg-color:${color.input.default.readOnly.bg};}input:checked[data-read-only]:disabled + &&{--switch-bg-color:${color.input.default.readOnly.fg};--switch-fg-color:${color.input.default.readOnly.bg};}`;
}
function switchTrackStyles(props) {
	let { input } = getTheme_v2(props.theme);
	return css`&:not([hidden]){display:block;}background-color:var(--switch-bg-color);position:absolute;left:0;top:0;width:${rem(input.switch.width)};height:${rem(input.switch.height)};border-radius:${rem(input.switch.height / 2)};`;
}
function switchThumbStyles(props) {
	let { $indeterminate } = props, { input } = getTheme_v2(props.theme), trackWidth = input.switch.width, trackHeight = input.switch.height, trackPadding = input.switch.padding, size = trackHeight - input.switch.padding * 2, checkedOffset = trackWidth - trackPadding * 2 - size, indeterminateOffset = trackWidth / 2 - size / 2 - trackPadding, checked = $indeterminate !== !0 && props.$checked === !0;
	return css`&:not([hidden]){display:block;}position:absolute;left:${rem(trackPadding)};top:${rem(trackPadding)};height:${rem(size)};width:${rem(size)};border-radius:${rem(size / 2)};transition-property:transform;transition-duration:${input.switch.transitionDurationMs}ms;transition-timing-function:${input.switch.transitionTimingFunction};background:var(--switch-fg-color);transform:translate3d(0,0,0);box-shadow:0px 1px 0px 0px rgba(0,0,0,0.05);${checked && css`transform:translate3d(${checkedOffset}px,0,0);`} ${$indeterminate && css`transform:translate3d(${indeterminateOffset}px,0,0);`}`;
}
const StyledSwitch = /* @__PURE__ */ styled.span.withConfig({
	displayName: "StyledSwitch",
	componentId: "sc-2t29u1-0"
})(switchBaseStyles), Input$1 = /* @__PURE__ */ styled.input.withConfig({
	displayName: "Input",
	componentId: "sc-2t29u1-1"
})(switchInputStyles), Representation = /* @__PURE__ */ styled.span.withConfig({
	displayName: "Representation",
	componentId: "sc-2t29u1-2"
})(switchRepresentationStyles), Track = /* @__PURE__ */ styled.span.withConfig({
	displayName: "Track",
	componentId: "sc-2t29u1-3"
})(switchTrackStyles), Thumb = /* @__PURE__ */ styled.span.withConfig({
	displayName: "Thumb",
	componentId: "sc-2t29u1-4"
})(switchThumbStyles), Switch = forwardRef(function(props, forwardedRef) {
	let $ = c(26), checked, className, disabled, indeterminate, readOnly, restProps, style;
	$[0] === props ? (checked = $[1], className = $[2], disabled = $[3], indeterminate = $[4], readOnly = $[5], restProps = $[6], style = $[7]) : ({checked, className, disabled, indeterminate, readOnly, style, ...restProps} = props, $[0] = props, $[1] = checked, $[2] = className, $[3] = disabled, $[4] = indeterminate, $[5] = readOnly, $[6] = restProps, $[7] = style);
	let ref = useRef(null), t0;
	$[8] === Symbol.for("react.memo_cache_sentinel") ? (t0 = () => ref.current, $[8] = t0) : t0 = $[8], useImperativeHandle(forwardedRef, t0);
	let t1, t2;
	$[9] === indeterminate ? (t1 = $[10], t2 = $[11]) : (t1 = () => {
		ref.current && (ref.current.indeterminate = indeterminate || !1);
	}, t2 = [indeterminate], $[9] = indeterminate, $[10] = t1, $[11] = t2), useEffect(t1, t2);
	let t3 = !disabled && readOnly ? "" : void 0, t4 = indeterminate !== !0 && checked, t5 = disabled || readOnly, t6;
	$[12] !== restProps || $[13] !== t3 || $[14] !== t4 || $[15] !== t5 ? (t6 = /* @__PURE__ */ jsx(Input$1, {
		"data-read-only": t3,
		...restProps,
		checked: t4,
		disabled: t5,
		type: "checkbox",
		ref
	}), $[12] = restProps, $[13] = t3, $[14] = t4, $[15] = t5, $[16] = t6) : t6 = $[16];
	let t7;
	$[17] === Symbol.for("react.memo_cache_sentinel") ? (t7 = /* @__PURE__ */ jsx(Track, {}), $[17] = t7) : t7 = $[17];
	let t8;
	$[18] !== checked || $[19] !== indeterminate ? (t8 = /* @__PURE__ */ jsxs(Representation, {
		"aria-hidden": !0,
		"data-name": "representation",
		children: [t7, /* @__PURE__ */ jsx(Thumb, {
			$checked: checked,
			$indeterminate: indeterminate
		})]
	}), $[18] = checked, $[19] = indeterminate, $[20] = t8) : t8 = $[20];
	let t9;
	return $[21] !== className || $[22] !== style || $[23] !== t6 || $[24] !== t8 ? (t9 = /* @__PURE__ */ jsxs(StyledSwitch, {
		className,
		"data-ui": "Switch",
		style,
		children: [t6, t8]
	}), $[21] = className, $[22] = style, $[23] = t6, $[24] = t8, $[25] = t9) : t9 = $[25], t9;
});
Switch.displayName = "ForwardRef(Switch)";
const StyledTextArea = /* @__PURE__ */ styled.span.withConfig({
	displayName: "StyledTextArea",
	componentId: "sc-5c9kno-0"
})(textInputRootStyle), InputRoot = styled.span.withConfig({
	displayName: "InputRoot",
	componentId: "sc-5c9kno-1"
})`flex:1;min-width:0;display:block;position:relative;`, Input = /* @__PURE__ */ styled.textarea.withConfig({
	displayName: "Input",
	componentId: "sc-5c9kno-2"
})(responsiveInputPaddingStyle, textInputBaseStyle, textInputFontSizeStyle), Presentation = /* @__PURE__ */ styled.div.withConfig({
	displayName: "Presentation",
	componentId: "sc-5c9kno-3"
})(responsiveRadiusStyle, textInputRepresentationStyle), TextArea = forwardRef(function(props, forwardedRef) {
	let $ = c(35), __unstable_disableFocusRing, customValidity, restProps, t0, t1, t2, t3, t4, weight;
	$[0] === props ? (__unstable_disableFocusRing = $[1], customValidity = $[2], restProps = $[3], t0 = $[4], t1 = $[5], t2 = $[6], t3 = $[7], t4 = $[8], weight = $[9]) : ({border: t0, customValidity, disabled: t1, fontSize: t2, padding: t3, radius: t4, weight, __unstable_disableFocusRing, ...restProps} = props, $[0] = props, $[1] = __unstable_disableFocusRing, $[2] = customValidity, $[3] = restProps, $[4] = t0, $[5] = t1, $[6] = t2, $[7] = t3, $[8] = t4, $[9] = weight);
	let border = t0 === void 0 ? !0 : t0, disabled = t1 === void 0 ? !1 : t1, fontSize = t2 === void 0 ? 2 : t2, padding = t3 === void 0 ? 3 : t3, radius = t4 === void 0 ? 2 : t4, ref = useRef(null), rootTheme = useRootTheme(), t5;
	$[10] === Symbol.for("react.memo_cache_sentinel") ? (t5 = () => ref.current, $[10] = t5) : t5 = $[10], useImperativeHandle(forwardedRef, t5), useCustomValidity(ref, customValidity);
	let t6 = rootTheme.scheme, t7 = rootTheme.tone, t8;
	$[11] === fontSize ? t8 = $[12] : (t8 = _getArrayProp(fontSize), $[11] = fontSize, $[12] = t8);
	let t9;
	$[13] === padding ? t9 = $[14] : (t9 = _getArrayProp(padding), $[13] = padding, $[14] = t9);
	let t10 = rootTheme.scheme, t11;
	$[15] === Symbol.for("react.memo_cache_sentinel") ? (t11 = _getArrayProp(0), $[15] = t11) : t11 = $[15];
	let t12;
	$[16] !== disabled || $[17] !== restProps || $[18] !== rootTheme.scheme || $[19] !== rootTheme.tone || $[20] !== t8 || $[21] !== t9 || $[22] !== weight ? (t12 = /* @__PURE__ */ jsx(Input, {
		"data-as": "textarea",
		"data-scheme": t6,
		"data-tone": t7,
		...restProps,
		$fontSize: t8,
		$padding: t9,
		$scheme: t10,
		$space: t11,
		$tone: rootTheme.tone,
		$weight: weight,
		disabled,
		ref
	}), $[16] = disabled, $[17] = restProps, $[18] = rootTheme.scheme, $[19] = rootTheme.tone, $[20] = t8, $[21] = t9, $[22] = weight, $[23] = t12) : t12 = $[23];
	let t13;
	$[24] === radius ? t13 = $[25] : (t13 = _getArrayProp(radius), $[24] = radius, $[25] = t13);
	let t14 = border ? "" : void 0, t15;
	$[26] !== __unstable_disableFocusRing || $[27] !== rootTheme.scheme || $[28] !== rootTheme.tone || $[29] !== t13 || $[30] !== t14 ? (t15 = /* @__PURE__ */ jsx(Presentation, {
		$radius: t13,
		$unstableDisableFocusRing: __unstable_disableFocusRing,
		$scheme: rootTheme.scheme,
		$tone: rootTheme.tone,
		"data-border": t14,
		"data-scheme": rootTheme.scheme,
		"data-tone": rootTheme.tone
	}), $[26] = __unstable_disableFocusRing, $[27] = rootTheme.scheme, $[28] = rootTheme.tone, $[29] = t13, $[30] = t14, $[31] = t15) : t15 = $[31];
	let t16;
	return $[32] !== t12 || $[33] !== t15 ? (t16 = /* @__PURE__ */ jsx(StyledTextArea, {
		"data-ui": "TextArea",
		children: /* @__PURE__ */ jsxs(InputRoot, { children: [t12, t15] })
	}), $[32] = t12, $[33] = t15, $[34] = t16) : t16 = $[34], t16;
});
TextArea.displayName = "ForwardRef(TextArea)";
/**
* @public
* Provides context for a group of tooltip elements that should share a delay
* which temporarily becomes 1 ms after the first floating element of the group opens.
*/
function TooltipDelayGroupProvider(props) {
	let $ = c(9), { children, delay } = props, [isGroupActive, setIsGroupActive] = useDelayedState(!1), [openTooltipId, setOpenTooltipId] = useDelayedState(null), openDelay = typeof delay == "number" ? delay : delay?.open || 0, closeDelay = typeof delay == "number" ? delay : delay?.close || 0, t0 = isGroupActive ? 1 : openDelay, t1;
	$[0] !== closeDelay || $[1] !== openTooltipId || $[2] !== setIsGroupActive || $[3] !== setOpenTooltipId || $[4] !== t0 ? (t1 = {
		setIsGroupActive,
		openTooltipId,
		setOpenTooltipId,
		openDelay: t0,
		closeDelay
	}, $[0] = closeDelay, $[1] = openTooltipId, $[2] = setIsGroupActive, $[3] = setOpenTooltipId, $[4] = t0, $[5] = t1) : t1 = $[5];
	let value = t1, t2;
	return $[6] !== children || $[7] !== value ? (t2 = /* @__PURE__ */ jsx(TooltipDelayGroupContext.Provider, {
		value,
		children
	}), $[6] = children, $[7] = value, $[8] = t2) : t2 = $[8], t2;
}
TooltipDelayGroupProvider.displayName = "TooltipDelayGroupProvider";
/**
* @internal
*/
const StyledAutocomplete = styled.div.withConfig({
	displayName: "StyledAutocomplete",
	componentId: "sc-890276-0"
})`line-height:0;`, ListBox = styled(Box).withConfig({
	displayName: "ListBox",
	componentId: "sc-890276-1"
})`& > ul{list-style:none;padding:0;margin:0;}`, rotate = keyframes`from{transform:rotate(0deg);}to{transform:rotate(360deg);}`, AnimatedSpinnerIcon = styled(SpinnerIcon).withConfig({
	displayName: "AnimatedSpinnerIcon",
	componentId: "sc-890276-2"
})`animation:${rotate} 500ms linear infinite;`;
function AutocompleteOption(props) {
	let $ = c(11), { children, id, onSelect, selected, value } = props, t0;
	$[0] !== onSelect || $[1] !== value ? (t0 = () => {
		setTimeout(() => {
			onSelect(value);
		}, 0);
	}, $[0] = onSelect, $[1] = value, $[2] = t0) : t0 = $[2];
	let handleClick = t0, t1;
	$[3] === handleClick ? t1 = $[4] : (t1 = (event) => {
		event.key === "Enter" && !_isEnterToClickElement(event.currentTarget) && handleClick();
	}, $[3] = handleClick, $[4] = t1);
	let handleKeyDown = t1, t2;
	return $[5] !== children || $[6] !== handleClick || $[7] !== handleKeyDown || $[8] !== id || $[9] !== selected ? (t2 = /* @__PURE__ */ jsx("li", {
		"aria-selected": selected,
		"data-ui": "AutocompleteOption",
		id,
		role: "option",
		onClick: handleClick,
		onKeyDown: handleKeyDown,
		children
	}), $[5] = children, $[6] = handleClick, $[7] = handleKeyDown, $[8] = id, $[9] = selected, $[10] = t2) : t2 = $[10], t2;
}
/**
* @internal
*/
function autocompleteReducer(state, msg) {
	return msg.type === "input/change" ? {
		...state,
		activeValue: null,
		focused: !0,
		query: msg.query
	} : msg.type === "input/focus" ? {
		...state,
		focused: !0
	} : msg.type === "root/blur" ? {
		...state,
		focused: !1,
		query: null
	} : msg.type === "root/clear" ? {
		...state,
		activeValue: null,
		query: null,
		value: null
	} : msg.type === "root/escape" ? {
		...state,
		focused: !1,
		query: null
	} : msg.type === "root/open" ? {
		...state,
		query: state.query || msg.query
	} : msg.type === "root/setActiveValue" ? {
		...state,
		activeValue: msg.value,
		listFocused: msg.listFocused || state.listFocused
	} : msg.type === "root/setListFocused" ? {
		...state,
		listFocused: msg.listFocused
	} : msg.type === "value/change" ? {
		...state,
		activeValue: msg.value,
		query: null,
		value: msg.value
	} : state;
}
/**
* @internal
*/
const AUTOCOMPLETE_LISTBOX_IGNORE_KEYS = [
	"Control",
	"Shift",
	"Alt",
	"Enter",
	"Home",
	"End",
	"PageUp",
	"PageDown",
	"Meta",
	"Tab",
	"CapsLock"
], AUTOCOMPLETE_POPOVER_FALLBACK_PLACEMENTS = ["bottom-start", "top-start"], DEFAULT_RENDER_VALUE = (value, option) => option ? option.value : value, DEFAULT_FILTER_OPTION = (query, option) => option.value.toLowerCase().indexOf(query.toLowerCase()) > -1, InnerAutocomplete = forwardRef(function(props, forwardedRef) {
	let $ = c(181), customValidity, disabled, filterOptionProp, icon, id, loading, onBlur, onChange, onFocus, onQueryChange, onSelect, openButton, openOnFocus, optionsProp, prefix, readOnly, relatedElements, renderOptionProp, renderPopover, restProps, suffix, t0, t1, t2, t3, t4, t5, t6, valueProp;
	$[0] === props ? (customValidity = $[1], disabled = $[2], filterOptionProp = $[3], icon = $[4], id = $[5], loading = $[6], onBlur = $[7], onChange = $[8], onFocus = $[9], onQueryChange = $[10], onSelect = $[11], openButton = $[12], openOnFocus = $[13], optionsProp = $[14], prefix = $[15], readOnly = $[16], relatedElements = $[17], renderOptionProp = $[18], renderPopover = $[19], restProps = $[20], suffix = $[21], t0 = $[22], t1 = $[23], t2 = $[24], t3 = $[25], t4 = $[26], t5 = $[27], t6 = $[28], valueProp = $[29]) : ({border: t0, customValidity, disabled, filterOption: filterOptionProp, fontSize: t1, icon, id, listBox: t2, loading, onBlur, onChange, onFocus, onQueryChange, onSelect, openButton, openOnFocus, options: optionsProp, padding: t3, popover: t4, prefix, radius: t5, readOnly, relatedElements, renderOption: renderOptionProp, renderPopover, renderValue: t6, suffix, value: valueProp, ...restProps} = props, $[0] = props, $[1] = customValidity, $[2] = disabled, $[3] = filterOptionProp, $[4] = icon, $[5] = id, $[6] = loading, $[7] = onBlur, $[8] = onChange, $[9] = onFocus, $[10] = onQueryChange, $[11] = onSelect, $[12] = openButton, $[13] = openOnFocus, $[14] = optionsProp, $[15] = prefix, $[16] = readOnly, $[17] = relatedElements, $[18] = renderOptionProp, $[19] = renderPopover, $[20] = restProps, $[21] = suffix, $[22] = t0, $[23] = t1, $[24] = t2, $[25] = t3, $[26] = t4, $[27] = t5, $[28] = t6, $[29] = valueProp);
	let border = t0 === void 0 ? !0 : t0, fontSize = t1 === void 0 ? 2 : t1, listBox = t2 === void 0 ? EMPTY_RECORD : t2, paddingProp = t3 === void 0 ? 3 : t3, popover = t4 === void 0 ? EMPTY_RECORD : t4, radius = t5 === void 0 ? 2 : t5, renderValue = t6 === void 0 ? DEFAULT_RENDER_VALUE : t6, t7 = valueProp || null, t8 = valueProp || null, t9;
	$[30] !== t7 || $[31] !== t8 ? (t9 = {
		activeValue: t7,
		focused: !1,
		listFocused: !1,
		query: null,
		value: t8
	}, $[30] = t7, $[31] = t8, $[32] = t9) : t9 = $[32];
	let [state, dispatch] = useReducer(autocompleteReducer, t9), { activeValue, focused, listFocused, query, value } = state, t10;
	$[33] !== fontSize || $[34] !== paddingProp ? (t10 = (t11) => {
		let { value: value_0 } = t11;
		return /* @__PURE__ */ jsx(Card, {
			"data-as": "button",
			padding: paddingProp,
			radius: 2,
			tone: "inherit",
			children: /* @__PURE__ */ jsx(Text, {
				size: fontSize,
				textOverflow: "ellipsis",
				children: value_0
			})
		});
	}, $[33] = fontSize, $[34] = paddingProp, $[35] = t10) : t10 = $[35];
	let renderOption = typeof renderOptionProp == "function" ? renderOptionProp : t10, filterOption = typeof filterOptionProp == "function" ? filterOptionProp : DEFAULT_FILTER_OPTION, rootElementRef = useRef(null), resultsPopoverElementRef = useRef(null), inputElementRef = useRef(null), listBoxElementRef = useRef(null), [inputElement, _setInputElement] = useState(null), t11;
	$[36] === Symbol.for("react.memo_cache_sentinel") ? (t11 = (node) => {
		startTransition(() => _setInputElement(node));
	}, $[36] = t11) : t11 = $[36];
	let setInputElement = t11, listFocusedRef = useRef(!1), valueRef = useRef(value), valuePropRef = useRef(valueProp), popoverMouseWithinRef = useRef(!1), t12, t13;
	$[37] === inputElement ? (t12 = $[38], t13 = $[39]) : (t12 = () => inputElement, t13 = [inputElement], $[37] = inputElement, $[38] = t12, $[39] = t13), useImperativeHandle(inputElementRef, t12, t13);
	let t14, t15;
	$[40] === inputElement ? (t14 = $[41], t15 = $[42]) : (t14 = () => inputElement, t15 = [inputElement], $[40] = inputElement, $[41] = t14, $[42] = t15), useImperativeHandle(forwardedRef, t14, t15);
	let listBoxId = `${id}-listbox`, options = Array.isArray(optionsProp) ? optionsProp : EMPTY_ARRAY, padding = _getArrayProp(paddingProp), t16;
	$[43] !== options || $[44] !== value ? (t16 = value === null ? void 0 : options.find((o) => o.value === value), $[43] = options, $[44] = value, $[45] = t16) : t16 = $[45];
	let currentOption = t16, t17;
	if ($[46] !== filterOption || $[47] !== options || $[48] !== query) {
		let t18;
		$[50] !== filterOption || $[51] !== query ? (t18 = (option) => query ? filterOption(query, option) : !0, $[50] = filterOption, $[51] = query, $[52] = t18) : t18 = $[52], t17 = options.filter(t18), $[46] = filterOption, $[47] = options, $[48] = query, $[49] = t17;
	} else t17 = $[49];
	let filteredOptions = t17, filteredOptionsLen = filteredOptions.length, activeItemId = activeValue ? `${id}-option-${activeValue}` : void 0, expanded = query !== null && loading || focused && query !== null, t18;
	$[53] !== onBlur || $[54] !== onQueryChange || $[55] !== relatedElements ? (t18 = (event) => {
		setTimeout(() => {
			if (popoverMouseWithinRef.current) return;
			let elements = (relatedElements || []).concat(rootElementRef.current ? [rootElementRef.current] : [], resultsPopoverElementRef.current ? [resultsPopoverElementRef.current] : []), focusInside = !1;
			if (document.activeElement) {
				for (let e of elements) if (e === document.activeElement || e.contains(document.activeElement)) {
					focusInside = !0;
					break;
				}
			}
			focusInside === !1 && (dispatch({ type: "root/blur" }), popoverMouseWithinRef.current = !1, onQueryChange && onQueryChange(null), onBlur && onBlur(event));
		}, 0);
	}, $[53] = onBlur, $[54] = onQueryChange, $[55] = relatedElements, $[56] = t18) : t18 = $[56];
	let handleRootBlur = t18, t19;
	$[57] === Symbol.for("react.memo_cache_sentinel") ? (t19 = (event_0) => {
		let listBoxElement = listBoxElementRef.current, focusedElement = event_0.target instanceof HTMLElement ? event_0.target : null, listFocused_0 = listBoxElement?.contains(focusedElement) || !1;
		listFocused_0 !== listFocusedRef.current && (listFocusedRef.current = listFocused_0, dispatch({
			type: "root/setListFocused",
			listFocused: listFocused_0
		}));
	}, $[57] = t19) : t19 = $[57];
	let handleRootFocus = t19, t20;
	$[58] !== onChange || $[59] !== onQueryChange || $[60] !== onSelect ? (t20 = (v) => {
		dispatch({
			type: "value/change",
			value: v
		}), popoverMouseWithinRef.current = !1, onSelect && onSelect(v), valueRef.current = v, onChange && onChange(v), onQueryChange && onQueryChange(null), inputElementRef.current?.focus();
	}, $[58] = onChange, $[59] = onQueryChange, $[60] = onSelect, $[61] = t20) : t20 = $[61];
	let handleOptionSelect = t20, t21;
	$[62] !== activeValue || $[63] !== filteredOptions || $[64] !== filteredOptionsLen || $[65] !== onQueryChange ? (t21 = (event_1) => {
		if (event_1.key === "ArrowDown") {
			if (event_1.preventDefault(), !filteredOptionsLen) return;
			let activeOption = filteredOptions.find((o_0) => o_0.value === activeValue), activeIndex = activeOption ? filteredOptions.indexOf(activeOption) : -1, nextActiveOption = filteredOptions[(activeIndex + 1) % filteredOptionsLen];
			nextActiveOption && dispatch({
				type: "root/setActiveValue",
				value: nextActiveOption.value,
				listFocused: !0
			});
			return;
		}
		if (event_1.key === "ArrowUp") {
			if (event_1.preventDefault(), !filteredOptionsLen) return;
			let activeOption_0 = filteredOptions.find((o_1) => o_1.value === activeValue), activeIndex_0 = activeOption_0 ? filteredOptions.indexOf(activeOption_0) : -1, nextActiveOption_0 = filteredOptions[activeIndex_0 === -1 ? filteredOptionsLen - 1 : (filteredOptionsLen + activeIndex_0 - 1) % filteredOptionsLen];
			nextActiveOption_0 && dispatch({
				type: "root/setActiveValue",
				value: nextActiveOption_0.value,
				listFocused: !0
			});
			return;
		}
		if (event_1.key === "Escape") {
			dispatch({ type: "root/escape" }), popoverMouseWithinRef.current = !1, onQueryChange && onQueryChange(null), inputElementRef.current?.focus();
			return;
		}
		let target = event_1.target, listEl = listBoxElementRef.current;
		if ((listEl === target || listEl?.contains(target)) && !AUTOCOMPLETE_LISTBOX_IGNORE_KEYS.includes(event_1.key)) {
			inputElementRef.current?.focus();
			return;
		}
	}, $[62] = activeValue, $[63] = filteredOptions, $[64] = filteredOptionsLen, $[65] = onQueryChange, $[66] = t21) : t21 = $[66];
	let handleRootKeyDown = t21, t22;
	$[67] === onQueryChange ? t22 = $[68] : (t22 = (event_2) => {
		let nextQuery = event_2.currentTarget.value;
		dispatch({
			type: "input/change",
			query: nextQuery
		}), onQueryChange && onQueryChange(nextQuery);
	}, $[67] = onQueryChange, $[68] = t22);
	let handleInputChange = t22, t23;
	$[69] !== currentOption || $[70] !== renderValue || $[71] !== value ? (t23 = () => {
		dispatch({
			type: "root/open",
			query: value ? renderValue(value, currentOption) : ""
		});
	}, $[69] = currentOption, $[70] = renderValue, $[71] = value, $[72] = t23) : t23 = $[72];
	let dispatchOpen = t23, t24;
	$[73] !== dispatchOpen || $[74] !== focused || $[75] !== onFocus || $[76] !== openOnFocus ? (t24 = (event_3) => {
		focused || (dispatch({ type: "input/focus" }), onFocus && onFocus(event_3), openOnFocus && dispatchOpen());
	}, $[73] = dispatchOpen, $[74] = focused, $[75] = onFocus, $[76] = openOnFocus, $[77] = t24) : t24 = $[77];
	let handleInputFocus = t24, t25;
	$[78] === Symbol.for("react.memo_cache_sentinel") ? (t25 = () => {
		popoverMouseWithinRef.current = !0;
	}, $[78] = t25) : t25 = $[78];
	let handlePopoverMouseEnter = t25, t26;
	$[79] === Symbol.for("react.memo_cache_sentinel") ? (t26 = () => {
		popoverMouseWithinRef.current = !1;
	}, $[79] = t26) : t26 = $[79];
	let handlePopoverMouseLeave = t26, t27;
	$[80] !== onChange || $[81] !== onQueryChange ? (t27 = () => {
		dispatch({ type: "root/clear" }), valueRef.current = "", onChange && onChange(""), onQueryChange && onQueryChange(null), inputElementRef.current?.focus();
	}, $[80] = onChange, $[81] = onQueryChange, $[82] = t27) : t27 = $[82];
	let handleClearButtonClick = t27, t28;
	$[83] === Symbol.for("react.memo_cache_sentinel") ? (t28 = () => {
		dispatch({ type: "input/focus" });
	}, $[83] = t28) : t28 = $[83];
	let handleClearButtonFocus = t28, t29, t30;
	$[84] === valueProp ? (t29 = $[85], t30 = $[86]) : (t29 = () => {
		if (valueProp !== valuePropRef.current) {
			valuePropRef.current = valueProp, valueProp !== void 0 && (dispatch({
				type: "value/change",
				value: valueProp
			}), valueRef.current = valueProp);
			return;
		}
		valueProp !== valueRef.current && (valueRef.current = valueProp || null, dispatch({
			type: "value/change",
			value: valueProp || null
		}));
	}, t30 = [valueProp], $[84] = valueProp, $[85] = t29, $[86] = t30), useEffect(t29, t30);
	let t31, t32;
	$[87] === focused ? (t31 = $[88], t32 = $[89]) : (t31 = () => {
		!focused && valueRef.current && dispatch({
			type: "root/setActiveValue",
			value: valueRef.current
		});
	}, t32 = [focused], $[87] = focused, $[88] = t31, $[89] = t32), useEffect(t31, t32);
	let t33, t34;
	$[90] !== activeValue || $[91] !== filteredOptions ? (t33 = () => {
		let listElement = listBoxElementRef.current;
		if (!listElement) return;
		let activeOption_1 = filteredOptions.find((o_2) => o_2.value === activeValue);
		if (activeOption_1) {
			let activeIndex_1 = filteredOptions.indexOf(activeOption_1), activeItemElement = listElement.childNodes[activeIndex_1];
			if (activeItemElement) {
				if (_hasFocus(activeItemElement)) return;
				focusFirstDescendant(activeItemElement);
			}
		}
	}, t34 = [activeValue, filteredOptions], $[90] = activeValue, $[91] = filteredOptions, $[92] = t33, $[93] = t34) : (t33 = $[92], t34 = $[93]), useEffect(t33, t34);
	let t35;
	bb0: {
		if (!loading && !disabled && value) {
			let t36;
			$[94] === Symbol.for("react.memo_cache_sentinel") ? (t36 = {
				"aria-label": "Clear",
				onFocus: handleClearButtonFocus
			}, $[94] = t36) : t36 = $[94], t35 = t36;
			break bb0;
		}
		t35 = void 0;
	}
	let clearButton = t35, openButtonBoxPadding = padding.map(_temp$2), openButtonPadding = padding.map(_temp2), openButtonProps = typeof openButton == "object" ? openButton : EMPTY_RECORD, t36;
	$[95] !== dispatchOpen || $[96] !== openButtonProps ? (t36 = (event_4) => {
		dispatchOpen(), openButtonProps.onClick && openButtonProps.onClick(event_4), _raf(() => inputElementRef.current?.focus());
	}, $[95] = dispatchOpen, $[96] = openButtonProps, $[97] = t36) : t36 = $[97];
	let handleOpenClick = t36, t37;
	$[98] !== disabled || $[99] !== expanded || $[100] !== fontSize || $[101] !== handleOpenClick || $[102] !== openButton || $[103] !== openButtonBoxPadding || $[104] !== openButtonPadding || $[105] !== openButtonProps || $[106] !== readOnly ? (t37 = !disabled && !readOnly && openButton ? /* @__PURE__ */ jsx(Box, {
		"aria-hidden": expanded,
		padding: openButtonBoxPadding,
		children: /* @__PURE__ */ jsx(Button, {
			"aria-label": "Open",
			disabled: expanded,
			fontSize,
			icon: ChevronDownIcon,
			mode: "bleed",
			padding: openButtonPadding,
			...openButtonProps,
			onClick: handleOpenClick
		})
	}) : void 0, $[98] = disabled, $[99] = expanded, $[100] = fontSize, $[101] = handleOpenClick, $[102] = openButton, $[103] = openButtonBoxPadding, $[104] = openButtonPadding, $[105] = openButtonProps, $[106] = readOnly, $[107] = t37) : t37 = $[107];
	let openButtonNode = t37, t38;
	bb1: {
		if (query === null) {
			if (value !== null) {
				let t39;
				$[108] !== currentOption || $[109] !== renderValue || $[110] !== value ? (t39 = renderValue(value, currentOption), $[108] = currentOption, $[109] = renderValue, $[110] = value, $[111] = t39) : t39 = $[111], t38 = t39;
				break bb1;
			}
			t38 = "";
			break bb1;
		}
		t38 = query;
	}
	let inputValue = t38, t39;
	$[112] === listFocused ? t39 = $[113] : (t39 = (event_5) => {
		event_5.key === "Tab" && listFocused && inputElementRef.current?.focus();
	}, $[112] = listFocused, $[113] = t39);
	let handleListBoxKeyDown = t39, t40;
	bb2: {
		if (filteredOptions.length === 0) {
			t40 = null;
			break bb2;
		}
		let t41;
		if ($[114] !== activeValue || $[115] !== currentOption || $[116] !== filteredOptions || $[117] !== handleOptionSelect || $[118] !== id || $[119] !== listFocused || $[120] !== loading || $[121] !== renderOption) {
			let t42;
			$[123] !== activeValue || $[124] !== currentOption || $[125] !== handleOptionSelect || $[126] !== id || $[127] !== listFocused || $[128] !== loading || $[129] !== renderOption ? (t42 = (option_0) => {
				let active = activeValue === null ? currentOption === option_0 : option_0.value === activeValue;
				return /* @__PURE__ */ jsx(AutocompleteOption, {
					id: `${id}-option-${option_0.value}`,
					onSelect: handleOptionSelect,
					selected: active,
					value: option_0.value,
					children: cloneElement(renderOption(option_0), {
						disabled: loading,
						selected: active,
						tabIndex: listFocused && active ? 0 : -1
					})
				}, option_0.value);
			}, $[123] = activeValue, $[124] = currentOption, $[125] = handleOptionSelect, $[126] = id, $[127] = listFocused, $[128] = loading, $[129] = renderOption, $[130] = t42) : t42 = $[130], t41 = filteredOptions.map(t42), $[114] = activeValue, $[115] = currentOption, $[116] = filteredOptions, $[117] = handleOptionSelect, $[118] = id, $[119] = listFocused, $[120] = loading, $[121] = renderOption, $[122] = t41;
		} else t41 = $[122];
		let t42;
		$[131] !== listBoxId || $[132] !== t41 ? (t42 = /* @__PURE__ */ jsx(Stack, {
			as: "ul",
			"aria-multiselectable": !1,
			"data-ui": "AutoComplete__resultsList",
			id: listBoxId,
			ref: listBoxElementRef,
			role: "listbox",
			space: 1,
			children: t41
		}), $[131] = listBoxId, $[132] = t41, $[133] = t42) : t42 = $[133];
		let t43;
		$[134] !== handleListBoxKeyDown || $[135] !== listBox || $[136] !== t42 ? (t43 = /* @__PURE__ */ jsx(ListBox, {
			"data-ui": "AutoComplete__results",
			onKeyDown: handleListBoxKeyDown,
			padding: 1,
			...listBox,
			tabIndex: -1,
			children: t42
		}), $[134] = handleListBoxKeyDown, $[135] = listBox, $[136] = t42, $[137] = t43) : t43 = $[137], t40 = t43;
	}
	let content = t40, t41;
	bb3: {
		if (renderPopover) {
			let t42 = !expanded, t43;
			$[138] !== content || $[139] !== handlePopoverMouseEnter || $[140] !== handlePopoverMouseLeave || $[141] !== inputElement || $[142] !== renderPopover || $[143] !== t42 ? (t43 = /* @__PURE__ */ jsx(RenderPopover, {
				content,
				hidden: t42,
				inputElement,
				onMouseEnter: handlePopoverMouseEnter,
				onMouseLeave: handlePopoverMouseLeave,
				resultsPopoverElementRef,
				renderPopover
			}), $[138] = content, $[139] = handlePopoverMouseEnter, $[140] = handlePopoverMouseLeave, $[141] = inputElement, $[142] = renderPopover, $[143] = t42, $[144] = t43) : t43 = $[144], t41 = t43;
			break bb3;
		}
		if (filteredOptionsLen === 0) {
			t41 = null;
			break bb3;
		}
		let t42;
		$[145] !== content || $[146] !== expanded || $[147] !== handlePopoverMouseEnter || $[148] !== handlePopoverMouseLeave || $[149] !== inputElement || $[150] !== popover || $[151] !== radius ? (t42 = /* @__PURE__ */ jsx(Popover, {
			arrow: !1,
			constrainSize: !0,
			content,
			fallbackPlacements: AUTOCOMPLETE_POPOVER_FALLBACK_PLACEMENTS,
			matchReferenceWidth: !0,
			onMouseEnter: handlePopoverMouseEnter,
			onMouseLeave: handlePopoverMouseLeave,
			open: expanded,
			overflow: "auto",
			placement: "bottom-start",
			portal: !0,
			radius,
			ref: resultsPopoverElementRef,
			referenceElement: inputElement,
			...popover
		}), $[145] = content, $[146] = expanded, $[147] = handlePopoverMouseEnter, $[148] = handlePopoverMouseLeave, $[149] = inputElement, $[150] = popover, $[151] = radius, $[152] = t42) : t42 = $[152], t41 = t42;
	}
	let results = t41, t42 = loading && AnimatedSpinnerIcon, t43 = suffix || openButtonNode, t44;
	$[153] !== activeItemId || $[154] !== border || $[155] !== clearButton || $[156] !== customValidity || $[157] !== disabled || $[158] !== expanded || $[159] !== fontSize || $[160] !== handleClearButtonClick || $[161] !== handleInputChange || $[162] !== handleInputFocus || $[163] !== icon || $[164] !== id || $[165] !== inputValue || $[166] !== listBoxId || $[167] !== padding || $[168] !== prefix || $[169] !== radius || $[170] !== readOnly || $[171] !== restProps || $[172] !== t42 || $[173] !== t43 ? (t44 = /* @__PURE__ */ jsx(TextInput, {
		...restProps,
		"aria-activedescendant": activeItemId,
		"aria-autocomplete": "list",
		"aria-expanded": expanded,
		"aria-owns": listBoxId,
		autoCapitalize: "off",
		autoComplete: "off",
		autoCorrect: "off",
		border,
		clearButton,
		customValidity,
		disabled,
		fontSize,
		icon,
		iconRight: t42,
		id,
		inputMode: "search",
		onChange: handleInputChange,
		onClear: handleClearButtonClick,
		onFocus: handleInputFocus,
		padding,
		prefix,
		radius,
		readOnly,
		ref: setInputElement,
		role: "combobox",
		spellCheck: !1,
		suffix: t43,
		value: inputValue
	}), $[153] = activeItemId, $[154] = border, $[155] = clearButton, $[156] = customValidity, $[157] = disabled, $[158] = expanded, $[159] = fontSize, $[160] = handleClearButtonClick, $[161] = handleInputChange, $[162] = handleInputFocus, $[163] = icon, $[164] = id, $[165] = inputValue, $[166] = listBoxId, $[167] = padding, $[168] = prefix, $[169] = radius, $[170] = readOnly, $[171] = restProps, $[172] = t42, $[173] = t43, $[174] = t44) : t44 = $[174];
	let t45;
	return $[175] !== handleRootBlur || $[176] !== handleRootFocus || $[177] !== handleRootKeyDown || $[178] !== results || $[179] !== t44 ? (t45 = /* @__PURE__ */ jsxs(StyledAutocomplete, {
		"data-ui": "Autocomplete",
		onBlur: handleRootBlur,
		onFocus: handleRootFocus,
		onKeyDown: handleRootKeyDown,
		ref: rootElementRef,
		children: [t44, results]
	}), $[175] = handleRootBlur, $[176] = handleRootFocus, $[177] = handleRootKeyDown, $[178] = results, $[179] = t44, $[180] = t45) : t45 = $[180], t45;
});
function RenderPopover({ renderPopover, content, hidden, inputElement, onMouseEnter, onMouseLeave, resultsPopoverElementRef }) {
	return renderPopover({
		content,
		hidden,
		inputElement,
		onMouseEnter,
		onMouseLeave
	}, resultsPopoverElementRef);
}
InnerAutocomplete.displayName = "ForwardRef(Autocomplete)";
/**
* The Autocomplete component is typically used for search components.
* It consists of a text input for writing a query, and properties for rendering suggestions.
*
* @public
*/
const Autocomplete = InnerAutocomplete;
function _temp$2(v_0) {
	return v_0 === 0 ? 0 : v_0 === 1 || v_0 === 2 ? 1 : v_0 - 2;
}
function _temp2(v_1) {
	return Math.max(v_1 - 1, 0);
}
const StyledBreadcrumbs = styled.ol.withConfig({
	displayName: "StyledBreadcrumbs",
	componentId: "sc-b0flr4-0"
})`margin:0;padding:0;display:flex;list-style:none;align-items:center;white-space:nowrap;line-height:0;`, ExpandButton = styled(Button).withConfig({
	displayName: "ExpandButton",
	componentId: "sc-b0flr4-1"
})`appearance:none;margin:-4px;`, Breadcrumbs = forwardRef(function(props, ref) {
	let $ = c(30), children, gap, maxLength, restProps, separator, t0;
	$[0] === props ? (children = $[1], gap = $[2], maxLength = $[3], restProps = $[4], separator = $[5], t0 = $[6]) : ({children, gap, maxLength, separator, space: t0, ...restProps} = props, $[0] = props, $[1] = children, $[2] = gap, $[3] = maxLength, $[4] = restProps, $[5] = separator, $[6] = t0);
	let t1 = gap === void 0 ? t0 === void 0 ? 2 : t0 : gap, t2;
	$[7] === t1 ? t2 = $[8] : (t2 = _getArrayProp(t1), $[7] = t1, $[8] = t2);
	let space = t2, [open, setOpen] = useState(!1), expandElementRef = useRef(null), popoverElementRef = useRef(null), t3;
	$[9] === Symbol.for("react.memo_cache_sentinel") ? (t3 = () => setOpen(!1), $[9] = t3) : t3 = $[9];
	let collapse = t3, t4;
	$[10] === Symbol.for("react.memo_cache_sentinel") ? (t4 = () => setOpen(!0), $[10] = t4) : t4 = $[10];
	let expand = t4, t5;
	$[11] === Symbol.for("react.memo_cache_sentinel") ? (t5 = () => [expandElementRef.current, popoverElementRef.current], $[11] = t5) : t5 = $[11], useClickOutsideEvent(collapse, t5);
	let t6;
	$[12] === children ? t6 = $[13] : (t6 = Children.toArray(children).filter(isValidElement), $[12] = children, $[13] = t6);
	let rawItems = t6, t7;
	$[14] !== maxLength || $[15] !== open || $[16] !== rawItems || $[17] !== space ? (t7 = {
		collapse,
		expand,
		expandElementRef,
		maxLength,
		open,
		popoverElementRef,
		rawItems,
		space
	}, $[14] = maxLength, $[15] = open, $[16] = rawItems, $[17] = space, $[18] = t7) : t7 = $[18];
	let items = useItems(t7), t8;
	if ($[19] !== items || $[20] !== separator || $[21] !== space) {
		let t9;
		$[23] !== separator || $[24] !== space ? (t9 = (item, itemIndex) => /* @__PURE__ */ jsxs(Fragment, { children: [itemIndex > 0 && /* @__PURE__ */ jsx(Box, {
			"aria-hidden": !0,
			as: "li",
			paddingX: space,
			children: separator || /* @__PURE__ */ jsx(Text, {
				muted: !0,
				children: "/"
			})
		}), /* @__PURE__ */ jsx(Box, {
			as: "li",
			children: item
		})] }, itemIndex), $[23] = separator, $[24] = space, $[25] = t9) : t9 = $[25], t8 = items.map(t9), $[19] = items, $[20] = separator, $[21] = space, $[22] = t8;
	} else t8 = $[22];
	let t9;
	return $[26] !== ref || $[27] !== restProps || $[28] !== t8 ? (t9 = /* @__PURE__ */ jsx(StyledBreadcrumbs, {
		"data-ui": "Breadcrumbs",
		...restProps,
		ref,
		children: t8
	}), $[26] = ref, $[27] = restProps, $[28] = t8, $[29] = t9) : t9 = $[29], t9;
});
Breadcrumbs.displayName = "ForwardRef(Breadcrumbs)";
function useItems(t0) {
	let $ = c(28), { collapse, expand, expandElementRef, maxLength, open, popoverElementRef, rawItems, space } = t0, len = rawItems.length;
	if (maxLength && len > maxLength) {
		let beforeLength = Math.ceil(maxLength / 2), afterLength = Math.floor(maxLength / 2), t1;
		if ($[0] !== afterLength || $[1] !== beforeLength || $[2] !== collapse || $[3] !== expand || $[4] !== expandElementRef || $[5] !== len || $[6] !== open || $[7] !== popoverElementRef || $[8] !== rawItems || $[9] !== space) {
			let t2 = rawItems.slice(0, beforeLength - 1), t3;
			$[11] !== afterLength || $[12] !== beforeLength || $[13] !== len || $[14] !== rawItems ? (t3 = rawItems.slice(beforeLength - 1, len - afterLength), $[11] = afterLength, $[12] = beforeLength, $[13] = len, $[14] = rawItems, $[15] = t3) : t3 = $[15];
			let t4;
			$[16] !== space || $[17] !== t3 ? (t4 = /* @__PURE__ */ jsx(Stack, {
				as: "ol",
				overflow: "auto",
				padding: space,
				gap: space,
				children: t3
			}), $[16] = space, $[17] = t3, $[18] = t4) : t4 = $[18];
			let t5 = open ? collapse : expand, t6;
			$[19] !== expandElementRef || $[20] !== open || $[21] !== t5 ? (t6 = /* @__PURE__ */ jsx(ExpandButton, {
				fontSize: 1,
				mode: "bleed",
				onClick: t5,
				padding: 1,
				ref: expandElementRef,
				selected: open,
				text: "…"
			}), $[19] = expandElementRef, $[20] = open, $[21] = t5, $[22] = t6) : t6 = $[22];
			let t7;
			$[23] !== open || $[24] !== popoverElementRef || $[25] !== t4 || $[26] !== t6 ? (t7 = /* @__PURE__ */ jsx(Popover, {
				constrainSize: !0,
				content: t4,
				open,
				placement: "top",
				portal: !0,
				ref: popoverElementRef,
				children: t6
			}, "button"), $[23] = open, $[24] = popoverElementRef, $[25] = t4, $[26] = t6, $[27] = t7) : t7 = $[27], t1 = [
				...t2,
				t7,
				...rawItems.slice(len - afterLength)
			], $[0] = afterLength, $[1] = beforeLength, $[2] = collapse, $[3] = expand, $[4] = expandElementRef, $[5] = len, $[6] = open, $[7] = popoverElementRef, $[8] = rawItems, $[9] = space, $[10] = t1;
		} else t1 = $[10];
		return t1;
	}
	return rawItems;
}
function dialogStyle({ theme }) {
	let { color } = getTheme_v2(theme);
	return {
		"&:not([hidden])": { display: "flex" },
		top: 0,
		left: 0,
		right: 0,
		bottom: 0,
		alignItems: "center",
		justifyContent: "center",
		outline: "none",
		background: color.backdrop
	};
}
function responsiveDialogPositionStyle(props) {
	let { media } = getTheme_v2(props.theme);
	return _responsive(media, props.$position, (position) => ({ "&&": { position } }));
}
function animationDialogStyle(props) {
	return props.$animate ? css`@keyframes zoomIn{from{opacity:0;transform:scale(0.95);}to{opacity:1;transform:scale(1);}}@keyframes fadeIn{from{opacity:0;}to{opacity:1;}}animation:fadeIn 200ms ease-out;& > [data-ui='DialogCard']{animation:zoomIn 200ms ease-out;}` : css``;
}
/**
* @internal
*/
const DialogContext = createGlobalScopedContext("@sanity/ui/context/dialog", { version: 0 });
/**
* This API might change. DO NOT USE IN PRODUCTION.
* @beta
*/
function useDialog() {
	return useContext(DialogContext);
}
function isTargetWithinScope(boundaryElement, portalElement, target) {
	return !boundaryElement || !portalElement ? !0 : containsOrEqualsElement(boundaryElement, target) || containsOrEqualsElement(portalElement, target);
}
const StyledDialog = /* @__PURE__ */ styled(Layer).withConfig({
	displayName: "StyledDialog",
	componentId: "sc-7l6kmb-0"
})(responsivePaddingStyle, dialogStyle, responsiveDialogPositionStyle, animationDialogStyle), DialogContainer = styled(Container).withConfig({
	displayName: "DialogContainer",
	componentId: "sc-7l6kmb-1"
})`&:not([hidden]){display:flex;}width:100%;height:100%;flex-direction:column;align-items:center;justify-content:center;`, DialogCardRoot = styled(Card).withConfig({
	displayName: "DialogCardRoot",
	componentId: "sc-7l6kmb-2"
})`&:not([hidden]){display:flex;}width:100%;min-height:0;max-height:100%;overflow:hidden;overflow:clip;`, DialogLayout = styled(Flex).withConfig({
	displayName: "DialogLayout",
	componentId: "sc-7l6kmb-3"
})`flex:1;min-height:0;width:100%;`, DialogHeader = styled(Box).withConfig({
	displayName: "DialogHeader",
	componentId: "sc-7l6kmb-4"
})`position:relative;z-index:2;`, DialogContent = styled(Box).withConfig({
	displayName: "DialogContent",
	componentId: "sc-7l6kmb-5"
})`position:relative;z-index:1;overflow:auto;outline:none;`, DialogFooter = styled(Box).withConfig({
	displayName: "DialogFooter",
	componentId: "sc-7l6kmb-6"
})`position:relative;z-index:3;`, DialogCard = forwardRef(function(props, forwardedRef) {
	let $ = c(44), { __unstable_autoFocus: autoFocus, __unstable_hideCloseButton: hideCloseButton, children, contentRef: forwardedContentRef, footer, header, id, onClickOutside, onClose, portal: portalProp, radius: radiusProp, scheme, shadow: shadowProp, width: widthProp } = props, portal = usePortal(), portalElement = portalProp ? portal.elements?.[portalProp] || null : portal.element, boundaryElement = useBoundaryElement().element, t0;
	$[0] === radiusProp ? t0 = $[1] : (t0 = _getArrayProp(radiusProp), $[0] = radiusProp, $[1] = t0);
	let radius = t0, t1;
	$[2] === shadowProp ? t1 = $[3] : (t1 = _getArrayProp(shadowProp), $[2] = shadowProp, $[3] = t1);
	let shadow = t1, t2;
	$[4] === widthProp ? t2 = $[5] : (t2 = _getArrayProp(widthProp), $[4] = widthProp, $[5] = t2);
	let width = t2, ref = useRef(null), contentRef = useRef(null), { isTopLayer } = useLayer(), labelId = `${id}_label`, showCloseButton = !!onClose && hideCloseButton === !1, showHeader = !!header || showCloseButton, t3;
	$[6] === Symbol.for("react.memo_cache_sentinel") ? (t3 = () => ref.current, $[6] = t3) : t3 = $[6], useImperativeHandle(forwardedRef, t3);
	let t4;
	$[7] === Symbol.for("react.memo_cache_sentinel") ? (t4 = () => contentRef.current, $[7] = t4) : t4 = $[7], useImperativeHandle(forwardedContentRef, t4);
	let t5, t6;
	$[8] === autoFocus ? (t5 = $[9], t6 = $[10]) : (t5 = () => {
		autoFocus && ref.current && focusFirstDescendant(ref.current);
	}, t6 = [autoFocus, ref], $[8] = autoFocus, $[9] = t5, $[10] = t6), useEffect(t5, t6);
	let t7;
	$[11] !== boundaryElement || $[12] !== isTopLayer || $[13] !== onClose || $[14] !== portalElement ? (t7 = (event) => {
		if (!isTopLayer || !onClose) return;
		let target = document.activeElement;
		target && !isTargetWithinScope(boundaryElement, portalElement, target) || event.key === "Escape" && (event.preventDefault(), event.stopPropagation(), onClose());
	}, $[11] = boundaryElement, $[12] = isTopLayer, $[13] = onClose, $[14] = portalElement, $[15] = t7) : t7 = $[15], useGlobalKeyDown(t7);
	let t8;
	$[16] !== boundaryElement || $[17] !== isTopLayer || $[18] !== onClickOutside || $[19] !== portalElement ? (t8 = isTopLayer && onClickOutside && ((event_0) => {
		let target_0 = event_0.target;
		target_0 && !isTargetWithinScope(boundaryElement, portalElement, target_0) || onClickOutside();
	}), $[16] = boundaryElement, $[17] = isTopLayer, $[18] = onClickOutside, $[19] = portalElement, $[20] = t8) : t8 = $[20];
	let t9;
	$[21] === Symbol.for("react.memo_cache_sentinel") ? (t9 = () => [ref.current], $[21] = t9) : t9 = $[21], useClickOutsideEvent(t8, t9);
	let t10;
	$[22] !== header || $[23] !== labelId || $[24] !== onClose || $[25] !== showCloseButton || $[26] !== showHeader ? (t10 = showHeader && /* @__PURE__ */ jsx(DialogHeader, { children: /* @__PURE__ */ jsxs(Flex, {
		align: "flex-start",
		padding: 3,
		children: [/* @__PURE__ */ jsx(Box, {
			flex: 1,
			padding: 2,
			children: header && /* @__PURE__ */ jsx(Text, {
				id: labelId,
				size: 1,
				weight: "semibold",
				children: header
			})
		}), showCloseButton && /* @__PURE__ */ jsx(Box, {
			flex: "none",
			children: /* @__PURE__ */ jsx(Button, {
				"aria-label": "Close dialog",
				disabled: !onClose,
				icon: CloseIcon,
				mode: "bleed",
				onClick: onClose,
				padding: 2
			})
		})]
	}) }), $[22] = header, $[23] = labelId, $[24] = onClose, $[25] = showCloseButton, $[26] = showHeader, $[27] = t10) : t10 = $[27];
	let t11;
	$[28] === children ? t11 = $[29] : (t11 = /* @__PURE__ */ jsx(DialogContent, {
		flex: 1,
		ref: contentRef,
		tabIndex: -1,
		children
	}), $[28] = children, $[29] = t11);
	let t12;
	$[30] === footer ? t12 = $[31] : (t12 = footer && /* @__PURE__ */ jsx(DialogFooter, { children: footer }), $[30] = footer, $[31] = t12);
	let t13;
	$[32] !== t10 || $[33] !== t11 || $[34] !== t12 ? (t13 = /* @__PURE__ */ jsxs(DialogLayout, {
		direction: "column",
		children: [
			t10,
			t11,
			t12
		]
	}), $[32] = t10, $[33] = t11, $[34] = t12, $[35] = t13) : t13 = $[35];
	let t14;
	$[36] !== radius || $[37] !== scheme || $[38] !== shadow || $[39] !== t13 ? (t14 = /* @__PURE__ */ jsx(DialogCardRoot, {
		radius,
		ref,
		scheme,
		shadow,
		children: t13
	}), $[36] = radius, $[37] = scheme, $[38] = shadow, $[39] = t13, $[40] = t14) : t14 = $[40];
	let t15;
	return $[41] !== t14 || $[42] !== width ? (t15 = /* @__PURE__ */ jsx(DialogContainer, {
		"data-ui": "DialogCard",
		width,
		children: t14
	}), $[41] = t14, $[42] = width, $[43] = t15) : t15 = $[43], t15;
});
DialogCard.displayName = "ForwardRef(DialogCard)";
/**
* The Dialog component.
*
* @public
*/
const Dialog = forwardRef(function(props, ref) {
	let $ = c(70), dialog = useDialog(), { layer } = useTheme_v2(), _positionProp, _zOffsetProp, children, contentRef, footer, header, id, onActivate, onClickOutside, onClose, onFocus, portalProp, restProps, scheme, t0, t1, t2, t3, t4, t5, t6;
	$[0] === props ? (_positionProp = $[1], _zOffsetProp = $[2], children = $[3], contentRef = $[4], footer = $[5], header = $[6], id = $[7], onActivate = $[8], onClickOutside = $[9], onClose = $[10], onFocus = $[11], portalProp = $[12], restProps = $[13], scheme = $[14], t0 = $[15], t1 = $[16], t2 = $[17], t3 = $[18], t4 = $[19], t5 = $[20], t6 = $[21]) : ({__unstable_autoFocus: t0, __unstable_hideCloseButton: t1, cardRadius: t2, cardShadow: t3, children, contentRef, footer, header, id, onActivate, onClickOutside, onClose, onFocus, padding: t4, portal: portalProp, position: _positionProp, scheme, width: t5, zOffset: _zOffsetProp, animate: t6, ...restProps} = props, $[0] = props, $[1] = _positionProp, $[2] = _zOffsetProp, $[3] = children, $[4] = contentRef, $[5] = footer, $[6] = header, $[7] = id, $[8] = onActivate, $[9] = onClickOutside, $[10] = onClose, $[11] = onFocus, $[12] = portalProp, $[13] = restProps, $[14] = scheme, $[15] = t0, $[16] = t1, $[17] = t2, $[18] = t3, $[19] = t4, $[20] = t5, $[21] = t6);
	let autoFocus = t0 === void 0 ? !0 : t0, hideCloseButton = t1 === void 0 ? !1 : t1, cardRadiusProp = t2 === void 0 ? 4 : t2, cardShadow = t3 === void 0 ? 3 : t3, paddingProp = t4 === void 0 ? 3 : t4, widthProp = t5 === void 0 ? 0 : t5, _animate = t6 === void 0 ? !1 : t6, positionProp = _positionProp ?? (dialog.position || "fixed"), zOffsetProp = _zOffsetProp ?? (dialog.zOffset || layer.dialog.zOffset), animate = usePrefersReducedMotion() ? !1 : _animate, portal = usePortal(), portalElement = portalProp ? portal.elements?.[portalProp] || null : portal.element, boundaryElement = useBoundaryElement().element, t7;
	$[22] === cardRadiusProp ? t7 = $[23] : (t7 = _getArrayProp(cardRadiusProp), $[22] = cardRadiusProp, $[23] = t7);
	let cardRadius = t7, t8;
	$[24] === paddingProp ? t8 = $[25] : (t8 = _getArrayProp(paddingProp), $[24] = paddingProp, $[25] = t8);
	let padding = t8, t9;
	$[26] === positionProp ? t9 = $[27] : (t9 = _getArrayProp(positionProp), $[26] = positionProp, $[27] = t9);
	let position = t9, t10;
	$[28] === widthProp ? t10 = $[29] : (t10 = _getArrayProp(widthProp), $[28] = widthProp, $[29] = t10);
	let width = t10, t11;
	$[30] === zOffsetProp ? t11 = $[31] : (t11 = _getArrayProp(zOffsetProp), $[30] = zOffsetProp, $[31] = t11);
	let zOffset = t11, preDivRef = useRef(null), postDivRef = useRef(null), cardRef = useRef(null), focusedElementRef = useRef(null), t12;
	$[32] === onFocus ? t12 = $[33] : (t12 = (event) => {
		onFocus?.(event);
		let target = event.target, cardElement = cardRef.current;
		if (cardElement && target === preDivRef.current) {
			focusLastDescendant(cardElement);
			return;
		}
		if (cardElement && target === postDivRef.current) {
			focusFirstDescendant(cardElement);
			return;
		}
		isHTMLElement(event.target) && (focusedElementRef.current = event.target);
	}, $[32] = onFocus, $[33] = t12);
	let handleFocus = t12, labelId = `${id}_label`, rootClickTimeoutRef = useRef(void 0), t13;
	$[34] !== boundaryElement || $[35] !== portalElement ? (t13 = () => {
		rootClickTimeoutRef.current && clearTimeout(rootClickTimeoutRef.current), rootClickTimeoutRef.current = setTimeout(() => {
			let activeElement = document.activeElement;
			if (activeElement && !isTargetWithinScope(boundaryElement, portalElement, activeElement)) {
				let target_0 = focusedElementRef.current;
				if (!target_0 || !document.body.contains(target_0)) {
					let cardElement_0 = cardRef.current;
					cardElement_0 && focusFirstDescendant(cardElement_0);
					return;
				}
				target_0.focus();
			}
		}, 0);
	}, $[34] = boundaryElement, $[35] = portalElement, $[36] = t13) : t13 = $[36];
	let handleRootClick = t13, t14;
	$[37] === Symbol.for("react.memo_cache_sentinel") ? (t14 = /* @__PURE__ */ jsx("div", {
		ref: preDivRef,
		tabIndex: 0
	}), $[37] = t14) : t14 = $[37];
	let t15;
	$[38] !== autoFocus || $[39] !== cardRadius || $[40] !== cardShadow || $[41] !== children || $[42] !== contentRef || $[43] !== footer || $[44] !== header || $[45] !== hideCloseButton || $[46] !== id || $[47] !== onClickOutside || $[48] !== onClose || $[49] !== portalProp || $[50] !== scheme || $[51] !== width ? (t15 = /* @__PURE__ */ jsx(DialogCard, {
		__unstable_autoFocus: autoFocus,
		__unstable_hideCloseButton: hideCloseButton,
		contentRef,
		footer,
		header,
		id,
		onClickOutside,
		onClose,
		portal: portalProp,
		radius: cardRadius,
		ref: cardRef,
		scheme,
		shadow: cardShadow,
		width,
		children
	}), $[38] = autoFocus, $[39] = cardRadius, $[40] = cardShadow, $[41] = children, $[42] = contentRef, $[43] = footer, $[44] = header, $[45] = hideCloseButton, $[46] = id, $[47] = onClickOutside, $[48] = onClose, $[49] = portalProp, $[50] = scheme, $[51] = width, $[52] = t15) : t15 = $[52];
	let t16;
	$[53] === Symbol.for("react.memo_cache_sentinel") ? (t16 = /* @__PURE__ */ jsx("div", {
		ref: postDivRef,
		tabIndex: 0
	}), $[53] = t16) : t16 = $[53];
	let t17;
	$[54] !== animate || $[55] !== handleFocus || $[56] !== handleRootClick || $[57] !== id || $[58] !== labelId || $[59] !== onActivate || $[60] !== padding || $[61] !== position || $[62] !== ref || $[63] !== restProps || $[64] !== t15 || $[65] !== zOffset ? (t17 = /* @__PURE__ */ jsxs(StyledDialog, {
		...restProps,
		$animate: animate,
		$padding: padding,
		$position: position,
		"aria-labelledby": labelId,
		"aria-modal": !0,
		"data-ui": "Dialog",
		id,
		onActivate,
		onClick: handleRootClick,
		onFocus: handleFocus,
		ref,
		role: "dialog",
		zOffset,
		children: [
			t14,
			t15,
			t16
		]
	}), $[54] = animate, $[55] = handleFocus, $[56] = handleRootClick, $[57] = id, $[58] = labelId, $[59] = onActivate, $[60] = padding, $[61] = position, $[62] = ref, $[63] = restProps, $[64] = t15, $[65] = zOffset, $[66] = t17) : t17 = $[66];
	let t18;
	return $[67] !== portalProp || $[68] !== t17 ? (t18 = /* @__PURE__ */ jsx(Portal, {
		__unstable_name: portalProp,
		children: t17
	}), $[67] = portalProp, $[68] = t17, $[69] = t18) : t18 = $[69], t18;
});
Dialog.displayName = "ForwardRef(Dialog)";
/**
* This API might change. DO NOT USE IN PRODUCTION.
* @beta
*/
function DialogProvider(props) {
	let $ = c(6), { children, position, zOffset } = props, t0;
	$[0] !== position || $[1] !== zOffset ? (t0 = {
		version: 0,
		position,
		zOffset
	}, $[0] = position, $[1] = zOffset, $[2] = t0) : t0 = $[2];
	let contextValue = t0, t1;
	return $[3] !== children || $[4] !== contextValue ? (t1 = /* @__PURE__ */ jsx(DialogContext.Provider, {
		value: contextValue,
		children
	}), $[3] = children, $[4] = contextValue, $[5] = t1) : t1 = $[5], t1;
}
DialogProvider.displayName = "DialogProvider";
/**
* The `MenuButton` component follows the WAI-ARIA specification for menu buttons.
*
* @public
*/
const MenuButton = forwardRef(function(props, forwardedRef) {
	let $ = c(62), { __unstable_disableRestoreFocusOnClose: t0, boundaryElement: deprecated_boundaryElement, button: buttonProp, id, menu: menuProp, onClose, onOpen, placement: deprecated_placement, popoverScheme: deprecated_popoverScheme, portal: t1, popover, popoverRadius: deprecated_popoverRadius, preventOverflow: deprecated_preventOverflow } = props, disableRestoreFocusOnClose = t0 === void 0 ? !1 : t0, deprecated_portal = t1 === void 0 ? !0 : t1, [open, setOpen] = useState(!1), [shouldFocus, setShouldFocus] = useState(null), [buttonElement, setButtonElement] = useState(null), t2;
	$[0] === Symbol.for("react.memo_cache_sentinel") ? (t2 = [], $[0] = t2) : t2 = $[0];
	let [menuElements, setChildMenuElements] = useState(t2), openRef = useRef(open), t3, t4;
	$[1] !== onOpen || $[2] !== open ? (t3 = () => {
		onOpen && open && !openRef.current && onOpen();
	}, t4 = [onOpen, open], $[1] = onOpen, $[2] = open, $[3] = t3, $[4] = t4) : (t3 = $[3], t4 = $[4]), useEffect(t3, t4);
	let t5, t6;
	$[5] !== onClose || $[6] !== open ? (t5 = () => {
		onClose && !open && openRef.current && onClose();
	}, t6 = [onClose, open], $[5] = onClose, $[6] = open, $[7] = t5, $[8] = t6) : (t5 = $[7], t6 = $[8]), useEffect(t5, t6);
	let t7, t8;
	$[9] === open ? (t7 = $[10], t8 = $[11]) : (t7 = () => {
		openRef.current = open;
	}, t8 = [open], $[9] = open, $[10] = t7, $[11] = t8), useEffect(t7, t8);
	let t9;
	$[12] === Symbol.for("react.memo_cache_sentinel") ? (t9 = () => {
		setOpen(_temp$1), setShouldFocus(null);
	}, $[12] = t9) : t9 = $[12];
	let handleButtonClick = t9, t10;
	$[13] === open ? t10 = $[14] : (t10 = (event) => {
		open && event.preventDefault();
	}, $[13] = open, $[14] = t10);
	let handleMouseDown = t10, t11;
	$[15] === Symbol.for("react.memo_cache_sentinel") ? (t11 = (event_0) => {
		if (event_0.key === "ArrowDown" || event_0.key === "Enter" || event_0.key === " ") {
			event_0.preventDefault(), setOpen(!0), setShouldFocus("first");
			return;
		}
		if (event_0.key === "ArrowUp") {
			event_0.preventDefault(), setOpen(!0), setShouldFocus("last");
			return;
		}
	}, $[15] = t11) : t11 = $[15];
	let handleButtonKeyDown = t11, t12;
	$[16] !== buttonElement || $[17] !== menuElements ? (t12 = (event_1) => {
		let target = event_1.target;
		if (target instanceof Node && !(buttonElement && (target === buttonElement || buttonElement.contains(target)))) {
			for (let el of menuElements) if (target === el || el.contains(target)) return;
			setOpen(!1);
		}
	}, $[16] = buttonElement, $[17] = menuElements, $[18] = t12) : t12 = $[18];
	let handleMenuClickOutside = t12, t13;
	$[19] !== buttonElement || $[20] !== disableRestoreFocusOnClose ? (t13 = () => {
		setOpen(!1), !disableRestoreFocusOnClose && buttonElement && buttonElement.focus();
	}, $[19] = buttonElement, $[20] = disableRestoreFocusOnClose, $[21] = t13) : t13 = $[21];
	let handleMenuEscape = t13, t14;
	$[22] === menuElements ? t14 = $[23] : (t14 = (event_2) => {
		let target_0 = event_2.relatedTarget;
		if (target_0 instanceof Node) {
			for (let el_0 of menuElements) if (el_0 === target_0 || el_0.contains(target_0)) return;
			setOpen(!1);
		}
	}, $[22] = menuElements, $[23] = t14);
	let handleBlur = t14, t15;
	$[24] !== buttonElement || $[25] !== disableRestoreFocusOnClose ? (t15 = () => {
		setOpen(!1), !disableRestoreFocusOnClose && buttonElement && buttonElement.focus();
	}, $[24] = buttonElement, $[25] = disableRestoreFocusOnClose, $[26] = t15) : t15 = $[26];
	let handleItemClick = t15, t16;
	$[27] === Symbol.for("react.memo_cache_sentinel") ? (t16 = (el_1) => (setChildMenuElements((els) => els.concat([el_1])), () => setChildMenuElements((els_0) => els_0.filter((_el) => _el !== el_1))), $[27] = t16) : t16 = $[27];
	let registerElement = t16, t17;
	$[28] !== buttonElement || $[29] !== handleBlur || $[30] !== handleItemClick || $[31] !== handleMenuClickOutside || $[32] !== handleMenuEscape || $[33] !== id || $[34] !== menuProp || $[35] !== shouldFocus ? (t17 = menuProp && cloneElement(menuProp, {
		"aria-labelledby": id,
		onBlurCapture: handleBlur,
		onClickOutside: handleMenuClickOutside,
		onEscape: handleMenuEscape,
		onItemClick: handleItemClick,
		originElement: buttonElement,
		registerElement,
		shouldFocus
	}), $[28] = buttonElement, $[29] = handleBlur, $[30] = handleItemClick, $[31] = handleMenuClickOutside, $[32] = handleMenuEscape, $[33] = id, $[34] = menuProp, $[35] = shouldFocus, $[36] = t17) : t17 = $[36];
	let menu = t17, t18;
	$[37] !== buttonProp || $[38] !== handleMouseDown || $[39] !== id || $[40] !== open ? (t18 = buttonProp && cloneElement(buttonProp, {
		"data-ui": "MenuButton",
		id,
		onClick: handleButtonClick,
		onKeyDown: handleButtonKeyDown,
		onMouseDown: handleMouseDown,
		"aria-haspopup": !0,
		"aria-expanded": open,
		ref: setButtonElement,
		selected: buttonProp.props.selected ?? open
	}), $[37] = buttonProp, $[38] = handleMouseDown, $[39] = id, $[40] = open, $[41] = t18) : t18 = $[41];
	let button = t18, t19, t20;
	$[42] === buttonElement ? (t19 = $[43], t20 = $[44]) : (t19 = () => buttonElement, t20 = [buttonElement], $[42] = buttonElement, $[43] = t19, $[44] = t20), useImperativeHandle(forwardedRef, t19, t20);
	let t21;
	$[45] === popover ? t21 = $[46] : (t21 = popover || {}, $[45] = popover, $[46] = t21);
	let t22;
	$[47] !== deprecated_boundaryElement || $[48] !== deprecated_placement || $[49] !== deprecated_popoverRadius || $[50] !== deprecated_popoverScheme || $[51] !== deprecated_portal || $[52] !== deprecated_preventOverflow || $[53] !== t21 ? (t22 = {
		boundaryElement: deprecated_boundaryElement,
		overflow: "auto",
		placement: deprecated_placement,
		portal: deprecated_portal,
		preventOverflow: deprecated_preventOverflow,
		radius: deprecated_popoverRadius,
		scheme: deprecated_popoverScheme,
		...t21
	}, $[47] = deprecated_boundaryElement, $[48] = deprecated_placement, $[49] = deprecated_popoverRadius, $[50] = deprecated_popoverScheme, $[51] = deprecated_portal, $[52] = deprecated_preventOverflow, $[53] = t21, $[54] = t22) : t22 = $[54];
	let popoverProps = t22, t23;
	$[55] === button ? t23 = $[56] : (t23 = button || /* @__PURE__ */ jsx(Fragment$1, {}), $[55] = button, $[56] = t23);
	let t24;
	return $[57] !== menu || $[58] !== open || $[59] !== popoverProps || $[60] !== t23 ? (t24 = /* @__PURE__ */ jsx(Popover, {
		"data-ui": "MenuButton__popover",
		...popoverProps,
		content: menu,
		open,
		children: t23
	}), $[57] = menu, $[58] = open, $[59] = popoverProps, $[60] = t23, $[61] = t24) : t24 = $[61], t24;
});
MenuButton.displayName = "ForwardRef(MenuButton)";
function _temp$1(v) {
	return !v;
}
const animation = css`background-image:linear-gradient( to right,var(--card-skeleton-color-from),var(--card-skeleton-color-to),var(--card-skeleton-color-from),var(--card-skeleton-color-from),var(--card-skeleton-color-from) );background-position:100%;background-size:200% 100%;background-attachment:fixed;animation-name:${keyframes`0%{background-position:100%;}100%{background-position:-100%;}`};animation-timing-function:ease-in-out;animation-iteration-count:infinite;animation-duration:2000ms;`, skeletonStyle = css`opacity:${({ $visible }) => +!!$visible};transition:opacity 200ms ease-in;@media screen and (prefers-reduced-motion:no-preference){${({ $animated }) => $animated ? animation : css`background-color:var(--card-skeleton-color-from);`}}@media screen and (prefers-reduced-motion:reduce){background-color:var(--card-skeleton-color-from);}`, StyledSkeleton$1 = /* @__PURE__ */ styled(Box).withConfig({
	displayName: "StyledSkeleton",
	componentId: "sc-zuj2vq-0"
})(responsiveRadiusStyle, skeletonStyle), Skeleton = forwardRef(function(props, ref) {
	let $ = c(16), delay, radius, restProps, t0;
	$[0] === props ? (delay = $[1], radius = $[2], restProps = $[3], t0 = $[4]) : ({animated: t0, delay, radius, ...restProps} = props, $[0] = props, $[1] = delay, $[2] = radius, $[3] = restProps, $[4] = t0);
	let animated = t0 === void 0 ? !1 : t0, [visible, setVisible] = useState(!delay), t1, t2;
	$[5] === delay ? (t1 = $[6], t2 = $[7]) : (t1 = () => {
		if (!delay) return;
		let timeout = setTimeout(() => {
			setVisible(!0);
		}, delay);
		return () => {
			clearTimeout(timeout);
		};
	}, t2 = [delay], $[5] = delay, $[6] = t1, $[7] = t2), useEffect(t1, t2);
	let t3;
	$[8] === radius ? t3 = $[9] : (t3 = _getArrayProp(radius), $[8] = radius, $[9] = t3);
	let t4 = delay ? visible : !0, t5;
	return $[10] !== animated || $[11] !== ref || $[12] !== restProps || $[13] !== t3 || $[14] !== t4 ? (t5 = /* @__PURE__ */ jsx(StyledSkeleton$1, {
		...restProps,
		$animated: animated,
		$radius: t3,
		$visible: t4,
		ref
	}), $[10] = animated, $[11] = ref, $[12] = restProps, $[13] = t3, $[14] = t4, $[15] = t5) : t5 = $[15], t5;
});
Skeleton.displayName = "ForwardRef(Skeleton)";
const StyledSkeleton = /* @__PURE__ */ styled(Skeleton).withConfig({
	displayName: "StyledSkeleton",
	componentId: "sc-9op362-0"
})((props) => {
	let { $size, $style } = props, { font, media } = getTheme_v2(props.theme), fontStyle = font[$style];
	return _responsive(media, $size, (sizeIndex) => {
		let fontSize = fontStyle.sizes[sizeIndex];
		return { height: fontSize.lineHeight - fontSize.ascenderHeight - fontSize.descenderHeight };
	});
}), TextSkeleton = forwardRef(function(props, ref) {
	let $ = c(9), restProps, t0;
	$[0] === props ? (restProps = $[1], t0 = $[2]) : ({size: t0, ...restProps} = props, $[0] = props, $[1] = restProps, $[2] = t0);
	let size = t0 === void 0 ? 2 : t0, t1;
	$[3] === size ? t1 = $[4] : (t1 = _getArrayProp(size), $[3] = size, $[4] = t1);
	let $size = t1, t2;
	return $[5] !== $size || $[6] !== ref || $[7] !== restProps ? (t2 = /* @__PURE__ */ jsx(StyledSkeleton, {
		...restProps,
		$size,
		ref,
		$style: "text"
	}), $[5] = $size, $[6] = ref, $[7] = restProps, $[8] = t2) : t2 = $[8], t2;
});
TextSkeleton.displayName = "ForwardRef(TextSkeleton)";
/**
* This API might change. DO NOT USE IN PRODUCTION.
* @beta
*/
const LabelSkeleton = forwardRef(function(props, ref) {
	let $ = c(9), restProps, t0;
	$[0] === props ? (restProps = $[1], t0 = $[2]) : ({size: t0, ...restProps} = props, $[0] = props, $[1] = restProps, $[2] = t0);
	let size = t0 === void 0 ? 2 : t0, t1;
	$[3] === size ? t1 = $[4] : (t1 = _getArrayProp(size), $[3] = size, $[4] = t1);
	let $size = t1, t2;
	return $[5] !== $size || $[6] !== ref || $[7] !== restProps ? (t2 = /* @__PURE__ */ jsx(StyledSkeleton, {
		...restProps,
		$size,
		ref,
		$style: "label"
	}), $[5] = $size, $[6] = ref, $[7] = restProps, $[8] = t2) : t2 = $[8], t2;
});
LabelSkeleton.displayName = "ForwardRef(LabelSkeleton)";
/**
* This API might change. DO NOT USE IN PRODUCTION.
* @beta
*/
const HeadingSkeleton = forwardRef(function(props, ref) {
	let $ = c(9), restProps, t0;
	$[0] === props ? (restProps = $[1], t0 = $[2]) : ({size: t0, ...restProps} = props, $[0] = props, $[1] = restProps, $[2] = t0);
	let size = t0 === void 0 ? 2 : t0, t1;
	$[3] === size ? t1 = $[4] : (t1 = _getArrayProp(size), $[3] = size, $[4] = t1);
	let $size = t1, t2;
	return $[5] !== $size || $[6] !== ref || $[7] !== restProps ? (t2 = /* @__PURE__ */ jsx(StyledSkeleton, {
		...restProps,
		$size,
		ref,
		$style: "heading"
	}), $[5] = $size, $[6] = ref, $[7] = restProps, $[8] = t2) : t2 = $[8], t2;
});
HeadingSkeleton.displayName = "ForwardRef(HeadingSkeleton)";
/**
* This API might change. DO NOT USE IN PRODUCTION.
* @beta
*/
const CodeSkeleton = forwardRef(function(props, ref) {
	let $ = c(9), restProps, t0;
	$[0] === props ? (restProps = $[1], t0 = $[2]) : ({size: t0, ...restProps} = props, $[0] = props, $[1] = restProps, $[2] = t0);
	let size = t0 === void 0 ? 2 : t0, t1;
	$[3] === size ? t1 = $[4] : (t1 = _getArrayProp(size), $[3] = size, $[4] = t1);
	let $size = t1, t2;
	return $[5] !== $size || $[6] !== ref || $[7] !== restProps ? (t2 = /* @__PURE__ */ jsx(StyledSkeleton, {
		...restProps,
		$size,
		ref,
		$style: "code"
	}), $[5] = $size, $[6] = ref, $[7] = restProps, $[8] = t2) : t2 = $[8], t2;
});
CodeSkeleton.displayName = "ForwardRef(CodeSkeleton)";
/**
* @public
*/
const TabPanel = forwardRef(function(props, ref) {
	let $ = c(9), flex, restProps;
	$[0] === props ? (flex = $[1], restProps = $[2]) : ({flex, ...restProps} = props, $[0] = props, $[1] = flex, $[2] = restProps);
	let t0 = props.tabIndex === void 0 ? 0 : props.tabIndex, t1;
	return $[3] !== flex || $[4] !== props.children || $[5] !== ref || $[6] !== restProps || $[7] !== t0 ? (t1 = /* @__PURE__ */ jsx(Box, {
		"data-ui": "TabPanel",
		...restProps,
		flex,
		ref,
		role: "tabpanel",
		tabIndex: t0,
		children: props.children
	}), $[3] = flex, $[4] = props.children, $[5] = ref, $[6] = restProps, $[7] = t0, $[8] = t1) : t1 = $[8], t1;
});
TabPanel.displayName = "ForwardRef(TabPanel)";
const STATUS_CARD_TONE = {
	error: "critical",
	warning: "caution",
	success: "positive",
	info: "neutral"
}, BUTTON_TONE = {
	error: "critical",
	warning: "caution",
	success: "positive",
	info: "neutral"
}, TextBox = styled(Flex).withConfig({
	displayName: "TextBox",
	componentId: "sc-wezd9c-0"
})`overflow-x:auto;`, StyledToast = styled(Card).withConfig({
	displayName: "StyledToast",
	componentId: "sc-wezd9c-1"
})`pointer-events:all;width:100%;position:relative;overflow:hidden;overflow:clip;&[data-has-duration]{padding-bottom:calc(${2}px / 2);}`, LoadingBar = styled.div.withConfig({
	displayName: "LoadingBar",
	componentId: "sc-wezd9c-2"
})`display:flex;position:absolute;bottom:0px;top:0px;left:0px;right:0px;pointer-events:none;z-index:-1;overflow:hidden;overflow:clip;background:transparent;align-items:flex-end;will-change:opacity;`, LoadingBarMask = styled(Card).withConfig({
	displayName: "LoadingBarMask",
	componentId: "sc-wezd9c-3"
})`position:absolute;top:0;left:-${2}px;right:-${2}px;bottom:${2}px;z-index:1;`, LoadingBarProgress = styled(Card).withConfig({
	displayName: "LoadingBarProgress",
	componentId: "sc-wezd9c-4"
})`display:block;height:100%;width:100%;transform-origin:0% 50%;background-color:${(props) => {
	let { color } = getTheme_v2(props.theme);
	return color.button.default[props.tone].enabled.bg;
}};`, ROLES = {
	error: "alert",
	warning: "alert",
	success: "alert",
	info: "alert"
};
/**
* The `Toast` component gives feedback to users when an action has taken place.
*
* Toasts can be closed with a close button, or auto-dismiss after a certain timeout.
*
* @public
*/
function Toast(props) {
	let $ = c(50), closable, description, duration, onClose, restProps, status, t0, title, updatedAt;
	$[0] === props ? (closable = $[1], description = $[2], duration = $[3], onClose = $[4], restProps = $[5], status = $[6], t0 = $[7], title = $[8], updatedAt = $[9]) : ({closable, description, duration, onClose, radius: t0, title, status, updatedAt, ...restProps} = props, $[0] = props, $[1] = closable, $[2] = description, $[3] = duration, $[4] = onClose, $[5] = restProps, $[6] = status, $[7] = t0, $[8] = title, $[9] = updatedAt);
	let radius = t0 === void 0 ? 3 : t0, cardTone = status ? STATUS_CARD_TONE[status] : "default", buttonTone = status ? BUTTON_TONE[status] : "default", role = status ? ROLES[status] : "status", visualDuration = usePrefersReducedMotion() ? 0 : .26, t1;
	$[10] === visualDuration ? t1 = $[11] : (t1 = visualDuration ? {
		type: "spring",
		visualDuration,
		bounce: .25
	} : { duration: 0 }, $[10] = visualDuration, $[11] = t1);
	let transition = t1, hasDuration = duration && isFinite(duration) && duration < 20736e5, t2;
	$[12] === Symbol.for("react.memo_cache_sentinel") ? (t2 = ["hidden", "initial"], $[12] = t2) : t2 = $[12];
	let initial = t2, t3;
	$[13] === Symbol.for("react.memo_cache_sentinel") ? (t3 = ["visible", "slideIn"], $[13] = t3) : t3 = $[13];
	let animate = t3, t4;
	$[14] === Symbol.for("react.memo_cache_sentinel") ? (t4 = ["hidden", "slideOut"], $[14] = t4) : t4 = $[14];
	let exit = t4, t5 = hasDuration ? "" : void 0, t6;
	$[15] === title ? t6 = $[16] : (t6 = title && /* @__PURE__ */ jsx(Text, {
		size: 1,
		weight: "medium",
		children: title
	}), $[15] = title, $[16] = t6);
	let t7;
	$[17] !== description || $[18] !== transition ? (t7 = description && /* @__PURE__ */ jsx(MotionText, {
		muted: !0,
		size: 1,
		variants: content,
		transition,
		children: description
	}), $[17] = description, $[18] = transition, $[19] = t7) : t7 = $[19];
	let t8;
	$[20] !== t6 || $[21] !== t7 ? (t8 = /* @__PURE__ */ jsx(TextBox, {
		flex: 1,
		padding: 3,
		children: /* @__PURE__ */ jsxs(Stack, {
			space: 3,
			children: [t6, t7]
		})
	}), $[20] = t6, $[21] = t7, $[22] = t8) : t8 = $[22];
	let t9;
	$[23] !== buttonTone || $[24] !== closable || $[25] !== onClose ? (t9 = closable && /* @__PURE__ */ jsx(Box, {
		padding: 1,
		children: /* @__PURE__ */ jsx(Button, {
			as: "button",
			icon: CloseIcon,
			mode: "bleed",
			padding: 2,
			tone: buttonTone,
			onClick: onClose,
			style: { verticalAlign: "top" }
		})
	}), $[23] = buttonTone, $[24] = closable, $[25] = onClose, $[26] = t9) : t9 = $[26];
	let t10;
	$[27] !== t8 || $[28] !== t9 || $[29] !== transition ? (t10 = /* @__PURE__ */ jsxs(MotionFlex, {
		align: "flex-start",
		variants: content,
		transition,
		children: [t8, t9]
	}), $[27] = t8, $[28] = t9, $[29] = transition, $[30] = t10) : t10 = $[30];
	let t11;
	$[31] !== cardTone || $[32] !== duration || $[33] !== hasDuration || $[34] !== onClose || $[35] !== radius || $[36] !== transition || $[37] !== updatedAt || $[38] !== visualDuration ? (t11 = hasDuration && /* @__PURE__ */ jsxs(MotionLoadingBar, {
		variants: content,
		transition,
		children: [/* @__PURE__ */ jsx(LoadingBarMask, {
			tone: cardTone,
			radius
		}), /* @__PURE__ */ jsx(MotionLoadingBarProgress, {
			tone: cardTone,
			initial: { scaleX: 0 },
			animate: { scaleX: 1 },
			transition: {
				delay: visualDuration,
				duration: duration / 1e3,
				ease: "linear"
			},
			onAnimationComplete: onClose
		}, `progress-${updatedAt}`)]
	}), $[31] = cardTone, $[32] = duration, $[33] = hasDuration, $[34] = onClose, $[35] = radius, $[36] = transition, $[37] = updatedAt, $[38] = visualDuration, $[39] = t11) : t11 = $[39];
	let t12;
	return $[40] !== cardTone || $[41] !== radius || $[42] !== restProps || $[43] !== role || $[44] !== t10 || $[45] !== t11 || $[46] !== t5 || $[47] !== transition || $[48] !== visualDuration ? (t12 = /* @__PURE__ */ jsxs(MotionToast, {
		"data-ui": "Toast",
		role,
		...restProps,
		"data-has-duration": t5,
		custom: visualDuration,
		radius,
		shadow: 2,
		tone: cardTone,
		forwardedAs: "li",
		layout: "position",
		variants: container,
		initial,
		animate,
		exit,
		transition,
		children: [t10, t11]
	}), $[40] = cardTone, $[41] = radius, $[42] = restProps, $[43] = role, $[44] = t10, $[45] = t11, $[46] = t5, $[47] = transition, $[48] = visualDuration, $[49] = t12) : t12 = $[49], t12;
}
Toast.displayName = "Toast";
const container = {
	initial: {
		y: 32,
		scale: .5,
		zIndex: 1
	},
	hidden: { opacity: 0 },
	visible: (visualDuration) => visualDuration ? {
		opacity: 1,
		transition: {
			when: "beforeChildren",
			staggerChildren: visualDuration / 3,
			duration: visualDuration / 3
		}
	} : { opacity: 1 },
	slideIn: {
		y: 0,
		scale: 1
	},
	slideOut: {
		zIndex: 0,
		scale: .75
	}
}, content = {
	initial: { willChange: "transform" },
	hidden: { opacity: 0 },
	visible: { opacity: 1 }
}, MotionToast = motion.create(StyledToast), MotionFlex = motion.create(Flex), MotionText = motion.create(Text), MotionLoadingBar = motion.create(LoadingBar), MotionLoadingBarProgress = motion.create(LoadingBarProgress);
/**
* Some components should only render after mounting to the DOM, and not be rendered at all during SSR renderToString or equivalent.
* @public
*/
function useMounted() {
	return useSyncExternalStore(subscribe, () => !0, () => !1);
}
const subscribe = () => () => {}, ToastContext = createGlobalScopedContext("@sanity/ui/context/toast", null);
/**
* @internal
*/
function ToastLayer(props) {
	let $ = c(9), { children, padding: t0, paddingX, paddingY, gap: t1 } = props, padding = t0 === void 0 ? 4 : t0, gap = t1 === void 0 ? 3 : t1, { zIndex } = useLayer(), t2;
	$[0] === zIndex ? t2 = $[1] : (t2 = { zIndex }, $[0] = zIndex, $[1] = t2);
	let t3;
	return $[2] !== children || $[3] !== gap || $[4] !== padding || $[5] !== paddingX || $[6] !== paddingY || $[7] !== t2 ? (t3 = /* @__PURE__ */ jsx(StyledLayer, {
		forwardedAs: "ul",
		"data-ui": "ToastProvider",
		padding,
		paddingX,
		paddingY,
		gap,
		columns: 1,
		style: t2,
		children
	}), $[2] = children, $[3] = gap, $[4] = padding, $[5] = paddingX, $[6] = paddingY, $[7] = t2, $[8] = t3) : t3 = $[8], t3;
}
ToastLayer.displayName = "ToastLayer";
const StyledLayer = styled(Grid).withConfig({
	displayName: "StyledLayer",
	componentId: "sc-8z432x-0"
})`box-sizing:border-box;position:fixed;right:0;bottom:0;list-style:none;pointer-events:none;max-width:420px;width:100%;`;
let toastId = 0;
/** @internal */
function generateToastId() {
	return String(toastId++);
}
/**
* @public
*/
function ToastProvider(props) {
	let $ = c(13), { children, padding, paddingX, paddingY, gap, zOffset: t0 } = props, zOffset = t0 === void 0 ? 1 : t0, t1;
	$[0] === Symbol.for("react.memo_cache_sentinel") ? (t1 = [], $[0] = t1) : t1 = $[0];
	let [state, setState] = useState(t1), mounted = useMounted(), t2;
	$[1] === Symbol.for("react.memo_cache_sentinel") ? (t2 = {
		version: 0,
		push: (params) => {
			let id = params.id || generateToastId(), duration = params.duration || 5e3;
			return startTransition(() => {
				setState((prevState) => {
					if (duration === .01) return prevState.filter((toast) => toast.id !== id);
					let dismiss = () => startTransition(() => setState((currentState) => currentState.filter((toast_0) => toast_0.id !== id)));
					return [...prevState.filter((toast_1) => toast_1.id !== id), {
						dismiss,
						id,
						updatedAt: Date.now(),
						params: {
							...params,
							duration
						}
					}];
				});
			}), id;
		}
	}, $[1] = t2) : t2 = $[1];
	let value = t2, t3;
	$[2] !== gap || $[3] !== mounted || $[4] !== padding || $[5] !== paddingX || $[6] !== paddingY || $[7] !== state || $[8] !== zOffset ? (t3 = mounted && /* @__PURE__ */ jsx(LayerProvider, {
		zOffset,
		children: /* @__PURE__ */ jsx(ToastLayer, {
			padding,
			paddingX,
			paddingY,
			gap,
			children: /* @__PURE__ */ jsx(AnimatePresence, {
				initial: !1,
				mode: "popLayout",
				children: state.map(_temp)
			})
		})
	}), $[2] = gap, $[3] = mounted, $[4] = padding, $[5] = paddingX, $[6] = paddingY, $[7] = state, $[8] = zOffset, $[9] = t3) : t3 = $[9];
	let t4;
	return $[10] !== children || $[11] !== t3 ? (t4 = /* @__PURE__ */ jsxs(ToastContext.Provider, {
		value,
		children: [children, t3]
	}), $[10] = children, $[11] = t3, $[12] = t4) : t4 = $[12], t4;
}
function _temp(t0) {
	let { dismiss: dismiss_0, id: id_0, params: params_0, updatedAt } = t0;
	return /* @__PURE__ */ jsx(Toast, {
		closable: params_0.closable,
		description: params_0.description,
		onClose: dismiss_0,
		status: params_0.status,
		title: params_0.title,
		duration: params_0.duration,
		updatedAt
	}, id_0);
}
ToastProvider.displayName = "ToastProvider";
/**
* @public
*/
function useToast() {
	let value = useContext(ToastContext);
	if (!value) throw Error("useToast(): missing context value");
	if (!isRecord(value) || value.version !== 0) throw Error("useToast(): the context value is not compatible");
	return value;
}
function _findPrevItemElement(state, itemElements, focusedElement) {
	let idx = itemElements.indexOf(focusedElement), els = itemElements.slice(0, idx), len = els.length;
	for (let i = len - 1; i >= 0; --i) {
		let itemKey = els[i].getAttribute("data-tree-key");
		if (!itemKey) continue;
		let segments = itemKey.split("/");
		segments.pop();
		let p = [], expanded = !0;
		for (let j = 0; j < segments.length; j += 1) if (p.push(segments[j]), !state[p.join("/")]?.expanded) {
			expanded = !1;
			break;
		}
		if (expanded) return els[i];
	}
	return null;
}
function _findNextItemElement(state, itemElements, focusedElement) {
	let idx = itemElements.indexOf(focusedElement), els = itemElements.slice(idx), len = itemElements.length;
	for (let i = 1; i < len; i += 1) {
		if (!els[i]) continue;
		let itemKey = els[i].getAttribute("data-tree-key");
		if (!itemKey) continue;
		let segments = itemKey.split("/");
		segments.pop();
		let p = [], expanded = !0;
		for (let j = 0; j < segments.length; j += 1) if (p.push(segments[j]), !state[p.join("/")]?.expanded) {
			expanded = !1;
			break;
		}
		if (expanded) return els[i];
	}
	return null;
}
function _focusItemElement(el) {
	if (el.getAttribute("role") === "treeitem" && el.focus(), el.getAttribute("role") === "none") {
		let firstChild = el.firstChild;
		firstChild && firstChild instanceof HTMLElement && firstChild.focus();
	}
}
/**
* @internal
*/
const TreeContext = createGlobalScopedContext("@sanity/ui/context/tree", null), Tree = forwardRef(function(props, forwardedRef) {
	let $ = c(38), children, gap, onFocus, restProps, t0;
	$[0] === props ? (children = $[1], gap = $[2], onFocus = $[3], restProps = $[4], t0 = $[5]) : ({children, gap, space: t0, onFocus, ...restProps} = props, $[0] = props, $[1] = children, $[2] = gap, $[3] = onFocus, $[4] = restProps, $[5] = t0);
	let spacing = gap === void 0 ? t0 === void 0 ? 1 : t0 : gap, ref = useRef(null), [focusedElement, setFocusedElement] = useState(null), focusedElementRef = useRef(focusedElement), t1;
	$[6] === Symbol.for("react.memo_cache_sentinel") ? (t1 = [], $[6] = t1) : t1 = $[6];
	let path = t1, t2;
	$[7] === Symbol.for("react.memo_cache_sentinel") ? (t2 = [], $[7] = t2) : t2 = $[7];
	let [itemElements, setItemElements] = useState(t2), t3;
	$[8] === Symbol.for("react.memo_cache_sentinel") ? (t3 = {}, $[8] = t3) : t3 = $[8];
	let [state, setState] = useState(t3), stateRef = useRef(state), t4;
	$[9] === Symbol.for("react.memo_cache_sentinel") ? (t4 = () => ref.current, $[9] = t4) : t4 = $[9], useImperativeHandle(forwardedRef, t4);
	let t5, t6;
	$[10] === focusedElement ? (t5 = $[11], t6 = $[12]) : (t5 = () => {
		focusedElementRef.current = focusedElement;
	}, t6 = [focusedElement], $[10] = focusedElement, $[11] = t5, $[12] = t6), useEffect(t5, t6);
	let t7, t8;
	$[13] === state ? (t7 = $[14], t8 = $[15]) : (t7 = () => {
		stateRef.current = state;
	}, t8 = [state], $[13] = state, $[14] = t7, $[15] = t8), useEffect(t7, t8);
	let t9;
	$[16] === Symbol.for("react.memo_cache_sentinel") ? (t9 = (element, path_0, expanded, selected) => (setState((s) => ({
		...s,
		[path_0]: {
			element,
			expanded
		}
	})), selected && setFocusedElement(element), () => {
		setState((s_0) => {
			let newState = { ...s_0 };
			return delete newState[path_0], newState;
		});
	}), $[16] = t9) : t9 = $[16];
	let registerItem = t9, t10;
	$[17] === Symbol.for("react.memo_cache_sentinel") ? (t10 = (path_1, expanded_0) => {
		setState((s_1) => {
			let itemState = s_1[path_1];
			return itemState ? {
				...s_1,
				[path_1]: {
					...itemState,
					expanded: expanded_0
				}
			} : s_1;
		});
	}, $[17] = t10) : t10 = $[17];
	let setExpanded = t10, t11 = focusedElement || itemElements[0] || null, t12;
	$[18] !== spacing || $[19] !== state || $[20] !== t11 ? (t12 = {
		version: 0,
		focusedElement: t11,
		level: 0,
		path,
		registerItem,
		setExpanded,
		setFocusedElement,
		gap: spacing,
		space: spacing,
		state
	}, $[18] = spacing, $[19] = state, $[20] = t11, $[21] = t12) : t12 = $[21];
	let contextValue = t12, t13;
	$[22] === itemElements ? t13 = $[23] : (t13 = (event) => {
		if (focusedElementRef.current) {
			if (event.key === "ArrowDown") {
				event.preventDefault();
				let nextEl = _findNextItemElement(stateRef.current, itemElements, focusedElementRef.current);
				nextEl && (_focusItemElement(nextEl), setFocusedElement(nextEl));
				return;
			}
			if (event.key === "ArrowUp") {
				event.preventDefault();
				let prevEl = _findPrevItemElement(stateRef.current, itemElements, focusedElementRef.current);
				prevEl && (_focusItemElement(prevEl), setFocusedElement(prevEl));
				return;
			}
			if (event.key === "ArrowLeft") {
				event.preventDefault();
				let itemKey = focusedElementRef.current.getAttribute("data-tree-key");
				if (!itemKey) return;
				let itemState_0 = stateRef.current[itemKey];
				if (!itemState_0) return;
				if (itemState_0.expanded) setState((s_2) => {
					let itemState_1 = s_2[itemKey];
					return itemState_1 ? {
						...s_2,
						[itemKey]: {
							...itemState_1,
							expanded: !1
						}
					} : s_2;
				});
				else {
					let itemPath = itemKey.split("/");
					itemPath.pop();
					let parentKey = itemPath.join("/"), parentState = parentKey && stateRef.current[parentKey];
					parentState && (parentState.element.focus(), setFocusedElement(parentState.element));
				}
				return;
			}
			if (event.key === "ArrowRight") {
				event.preventDefault();
				let focusedKey = focusedElementRef.current.getAttribute("data-tree-key");
				if (!focusedKey) return;
				stateRef.current[focusedKey]?.expanded || setState((s_3) => {
					let itemState_2 = s_3[focusedKey];
					return itemState_2 ? {
						...s_3,
						[focusedKey]: {
							...itemState_2,
							expanded: !0
						}
					} : s_3;
				});
				return;
			}
		}
	}, $[22] = itemElements, $[23] = t13);
	let handleKeyDown = t13, t14;
	$[24] === onFocus ? t14 = $[25] : (t14 = (event_0) => {
		setFocusedElement(event_0.target), onFocus?.(event_0);
	}, $[24] = onFocus, $[25] = t14);
	let handleFocus = t14, t15;
	$[26] === Symbol.for("react.memo_cache_sentinel") ? (t15 = () => {
		if (!ref.current) return;
		let _itemElements = Array.from(ref.current.querySelectorAll("[data-ui=\"TreeItem\"]"));
		setItemElements(_itemElements);
	}, $[26] = t15) : t15 = $[26];
	let t16;
	$[27] === children ? t16 = $[28] : (t16 = [children], $[27] = children, $[28] = t16), useEffect(t15, t16);
	let t17;
	$[29] !== children || $[30] !== handleFocus || $[31] !== handleKeyDown || $[32] !== restProps || $[33] !== spacing ? (t17 = /* @__PURE__ */ jsx(Stack, {
		as: "ul",
		"data-ui": "Tree",
		...restProps,
		onFocus: handleFocus,
		onKeyDown: handleKeyDown,
		ref,
		role: "tree",
		gap: spacing,
		children
	}), $[29] = children, $[30] = handleFocus, $[31] = handleKeyDown, $[32] = restProps, $[33] = spacing, $[34] = t17) : t17 = $[34];
	let t18;
	return $[35] !== contextValue || $[36] !== t17 ? (t18 = /* @__PURE__ */ jsx(TreeContext.Provider, {
		value: contextValue,
		children: t17
	}), $[35] = contextValue, $[36] = t17, $[37] = t18) : t18 = $[37], t18;
});
Tree.displayName = "ForwardRef(Tree)";
function treeItemRootStyle() {
	return css`&[role='none'] > [role='treeitem']{outline:none;cursor:default;border-radius:3px;background-color:var(--card-bg-color);color:var(--treeitem-fg-color);&:focus{position:relative;}}&[role='treeitem']{outline:none;& > div{cursor:default;border-radius:3px;background-color:var(--card-bg-color);color:var(--treeitem-fg-color);}&:focus > div{position:relative;}}`;
}
function treeItemRootColorStyle(props) {
	let { color } = getTheme_v2(props.theme), tone = color.selectable.default;
	return css`&[role='none']{& > [role='treeitem']{${_cardColorStyle(color, tone.enabled)}}&[data-selected] > [role='treeitem']{${_cardColorStyle(color, tone.pressed)}}@media (hover:hover){&:not([data-selected]) > [role='treeitem']:not(:focus):hover{${_cardColorStyle(color, tone.hovered)}}& > [role='treeitem']:focus{${_cardColorStyle(color, tone.selected)}}}}&[role='treeitem']{& > [data-ui='TreeItem__box']{${_cardColorStyle(color, tone.enabled)}}&[data-selected] > [data-ui='TreeItem__box']{${_cardColorStyle(color, tone.pressed)}}@media (hover:hover){&:not([data-selected]):not(:focus) > [data-ui='TreeItem__box']:hover{${_cardColorStyle(color, tone.hovered)}}&:focus > [data-ui='TreeItem__box']{${_cardColorStyle(color, tone.selected)}}}}`;
}
function treeItemBoxStyle(props) {
	let { $level } = props, { space } = getTheme_v2(props.theme);
	return css`padding-left:${rem(space[2] * $level)};&[data-as='a']{text-decoration:none;}`;
}
/**
* @beta
*/
function useTree() {
	let tree = useContext(TreeContext);
	if (!tree) throw Error("Tree: missing context value");
	return tree;
}
function TreeGroup(props) {
	let $ = c(9), children, restProps, t0;
	$[0] === props ? (children = $[1], restProps = $[2], t0 = $[3]) : ({children, expanded: t0, ...restProps} = props, $[0] = props, $[1] = children, $[2] = restProps, $[3] = t0);
	let expanded = t0 === void 0 ? !1 : t0, tree = useTree(), t1 = !expanded, t2;
	return $[4] !== children || $[5] !== restProps || $[6] !== t1 || $[7] !== tree.gap ? (t2 = /* @__PURE__ */ jsx(Stack, {
		as: "ul",
		"data-ui": "TreeGroup",
		...restProps,
		hidden: t1,
		marginTop: tree.gap,
		role: "group",
		gap: tree.gap,
		children
	}), $[4] = children, $[5] = restProps, $[6] = t1, $[7] = tree.gap, $[8] = t2) : t2 = $[8], t2;
}
const StyledTreeItem = /* @__PURE__ */ styled.li.withConfig({
	displayName: "StyledTreeItem",
	componentId: "sc-v9loy0-0"
})(treeItemRootStyle, treeItemRootColorStyle), TreeItemBox = /* @__PURE__ */ styled(Box).attrs({ forwardedAs: "a" }).withConfig({
	displayName: "TreeItemBox",
	componentId: "sc-v9loy0-1"
})(treeItemBoxStyle), ToggleArrowText = styled(Text).withConfig({
	displayName: "ToggleArrowText",
	componentId: "sc-v9loy0-2"
})`& > svg{transition:transform 100ms;}`;
/**
* This API might change. DO NOT USE IN PRODUCTION.
* @beta
*/
function TreeItem(props) {
	let $ = c(114), IconComponent, children, gap, href, idProp, linkAs, muted, onClick, restProps, t0, t1, t2, t3, t4, text, weight;
	$[0] === props ? (IconComponent = $[1], children = $[2], gap = $[3], href = $[4], idProp = $[5], linkAs = $[6], muted = $[7], onClick = $[8], restProps = $[9], t0 = $[10], t1 = $[11], t2 = $[12], t3 = $[13], t4 = $[14], text = $[15], weight = $[16]) : ({children, expanded: t0, fontSize: t1, href, icon: IconComponent, id: idProp, linkAs, muted, onClick, padding: t2, selected: t3, gap, space: t4, text, weight, ...restProps} = props, $[0] = props, $[1] = IconComponent, $[2] = children, $[3] = gap, $[4] = href, $[5] = idProp, $[6] = linkAs, $[7] = muted, $[8] = onClick, $[9] = restProps, $[10] = t0, $[11] = t1, $[12] = t2, $[13] = t3, $[14] = t4, $[15] = text, $[16] = weight);
	let expandedProp = t0 === void 0 ? !1 : t0, fontSize = t1 === void 0 ? 1 : t1, padding = t2 === void 0 ? 2 : t2, selected = t3 === void 0 ? !1 : t3, spacing = gap === void 0 ? t4 === void 0 ? 2 : t4 : gap, [rootElement, _setRootElement] = useState(null), t5;
	$[17] === Symbol.for("react.memo_cache_sentinel") ? (t5 = (node) => {
		startTransition(() => _setRootElement(node));
	}, $[17] = t5) : t5 = $[17];
	let setRootElement = t5, treeitemRef = useRef(null), tree = useTree(), { path, registerItem, setExpanded, setFocusedElement } = tree, _id = useId(), id = idProp || _id, t6, t7;
	if ($[18] !== id || $[19] !== path) {
		let itemPath = path.concat([id || ""]);
		t6 = itemPath, t7 = itemPath.join("/"), $[18] = id, $[19] = path, $[20] = t6, $[21] = t7;
	} else t6 = $[20], t7 = $[21];
	let t8;
	$[22] !== t6 || $[23] !== t7 ? (t8 = [t6, t7], $[22] = t6, $[23] = t7, $[24] = t8) : t8 = $[24];
	let [itemPath_0, itemKey] = t8, itemState = tree.state[itemKey], focused = tree.focusedElement === rootElement, expanded = itemState?.expanded === void 0 ? expandedProp : itemState?.expanded || !1, tabIndex = tree.focusedElement && tree.focusedElement === rootElement ? 0 : -1, t9 = tree.level + 1, t10;
	$[25] !== itemPath_0 || $[26] !== t9 || $[27] !== tree ? (t10 = {
		...tree,
		level: t9,
		path: itemPath_0
	}, $[25] = itemPath_0, $[26] = t9, $[27] = tree, $[28] = t10) : t10 = $[28];
	let contextValue = t10, t11;
	$[29] !== expanded || $[30] !== itemKey || $[31] !== onClick || $[32] !== rootElement || $[33] !== setExpanded || $[34] !== setFocusedElement ? (t11 = (event) => {
		onClick && onClick(event);
		let target = event.target;
		target instanceof HTMLElement && (target.getAttribute("data-ui") === "TreeItem" || target.closest("[data-ui=\"TreeItem__box\"]")) && (event.stopPropagation(), setExpanded(itemKey, !expanded), setFocusedElement(rootElement));
	}, $[29] = expanded, $[30] = itemKey, $[31] = onClick, $[32] = rootElement, $[33] = setExpanded, $[34] = setFocusedElement, $[35] = t11) : t11 = $[35];
	let handleClick = t11, t12;
	$[36] !== focused || $[37] !== rootElement ? (t12 = (event_0) => {
		focused && event_0.key === "Enter" && (treeitemRef.current || rootElement)?.click();
	}, $[36] = focused, $[37] = rootElement, $[38] = t12) : t12 = $[38];
	let handleKeyDown = t12, t13, t14;
	$[39] !== expanded || $[40] !== itemKey || $[41] !== registerItem || $[42] !== rootElement || $[43] !== selected ? (t13 = () => {
		if (rootElement) return registerItem(rootElement, itemKey, expanded, selected);
	}, t14 = [
		expanded,
		itemKey,
		registerItem,
		rootElement,
		selected
	], $[39] = expanded, $[40] = itemKey, $[41] = registerItem, $[42] = rootElement, $[43] = selected, $[44] = t13, $[45] = t14) : (t13 = $[44], t14 = $[45]), useEffect(t13, t14);
	let t15 = IconComponent || children ? "visible" : "hidden", t16;
	$[46] === t15 ? t16 = $[47] : (t16 = {
		visibility: t15,
		pointerEvents: "none"
	}, $[46] = t15, $[47] = t16);
	let t17;
	$[48] !== IconComponent || $[49] !== fontSize || $[50] !== muted || $[51] !== weight ? (t17 = IconComponent && /* @__PURE__ */ jsx(Text, {
		muted,
		size: fontSize,
		weight,
		children: /* @__PURE__ */ jsx(IconComponent, {})
	}), $[48] = IconComponent, $[49] = fontSize, $[50] = muted, $[51] = weight, $[52] = t17) : t17 = $[52];
	let t18;
	$[53] !== IconComponent || $[54] !== expanded || $[55] !== fontSize || $[56] !== muted || $[57] !== weight ? (t18 = !IconComponent && /* @__PURE__ */ jsx(ToggleArrowText, {
		muted,
		size: fontSize,
		weight,
		children: /* @__PURE__ */ jsx(ToggleArrowRightIcon, { style: { transform: expanded ? "rotate(90deg)" : void 0 } })
	}), $[53] = IconComponent, $[54] = expanded, $[55] = fontSize, $[56] = muted, $[57] = weight, $[58] = t18) : t18 = $[58];
	let t19;
	$[59] !== spacing || $[60] !== t16 || $[61] !== t17 || $[62] !== t18 ? (t19 = /* @__PURE__ */ jsxs(Box, {
		marginRight: spacing,
		style: t16,
		children: [t17, t18]
	}), $[59] = spacing, $[60] = t16, $[61] = t17, $[62] = t18, $[63] = t19) : t19 = $[63];
	let t20;
	$[64] !== fontSize || $[65] !== muted || $[66] !== text || $[67] !== weight ? (t20 = /* @__PURE__ */ jsx(Box, {
		flex: 1,
		children: /* @__PURE__ */ jsx(Text, {
			muted,
			size: fontSize,
			textOverflow: "ellipsis",
			weight,
			children: text
		})
	}), $[64] = fontSize, $[65] = muted, $[66] = text, $[67] = weight, $[68] = t20) : t20 = $[68];
	let t21;
	$[69] !== padding || $[70] !== t19 || $[71] !== t20 ? (t21 = /* @__PURE__ */ jsxs(Flex, {
		padding,
		children: [t19, t20]
	}), $[69] = padding, $[70] = t19, $[71] = t20, $[72] = t21) : t21 = $[72];
	let content = t21;
	if (href) {
		let t22 = selected ? "" : void 0, t23;
		$[73] !== content || $[74] !== expanded || $[75] !== href || $[76] !== linkAs || $[77] !== tabIndex || $[78] !== tree.level ? (t23 = /* @__PURE__ */ jsx(TreeItemBox, {
			$level: tree.level,
			"aria-expanded": expanded,
			as: linkAs,
			"data-ui": "TreeItem__box",
			href,
			ref: treeitemRef,
			role: "treeitem",
			tabIndex,
			children: content
		}), $[73] = content, $[74] = expanded, $[75] = href, $[76] = linkAs, $[77] = tabIndex, $[78] = tree.level, $[79] = t23) : t23 = $[79];
		let t24;
		$[80] !== children || $[81] !== expanded ? (t24 = children && /* @__PURE__ */ jsx(TreeGroup, {
			hidden: !expanded,
			children
		}), $[80] = children, $[81] = expanded, $[82] = t24) : t24 = $[82];
		let t25;
		$[83] !== contextValue || $[84] !== t24 ? (t25 = /* @__PURE__ */ jsx(TreeContext.Provider, {
			value: contextValue,
			children: t24
		}), $[83] = contextValue, $[84] = t24, $[85] = t25) : t25 = $[85];
		let t26;
		return $[86] !== handleClick || $[87] !== id || $[88] !== itemKey || $[89] !== restProps || $[90] !== t22 || $[91] !== t23 || $[92] !== t25 ? (t26 = /* @__PURE__ */ jsxs(StyledTreeItem, {
			"data-selected": t22,
			"data-tree-id": id,
			"data-tree-key": itemKey,
			"data-ui": "TreeItem",
			...restProps,
			onClick: handleClick,
			ref: setRootElement,
			role: "none",
			children: [t23, t25]
		}), $[86] = handleClick, $[87] = id, $[88] = itemKey, $[89] = restProps, $[90] = t22, $[91] = t23, $[92] = t25, $[93] = t26) : t26 = $[93], t26;
	}
	let t22 = selected ? "" : void 0, t23;
	$[94] !== content || $[95] !== tree.level ? (t23 = /* @__PURE__ */ jsx(TreeItemBox, {
		$level: tree.level,
		as: "div",
		"data-ui": "TreeItem__box",
		children: content
	}), $[94] = content, $[95] = tree.level, $[96] = t23) : t23 = $[96];
	let t24;
	$[97] !== children || $[98] !== expanded ? (t24 = children && /* @__PURE__ */ jsx(TreeGroup, {
		expanded,
		children
	}), $[97] = children, $[98] = expanded, $[99] = t24) : t24 = $[99];
	let t25;
	$[100] !== contextValue || $[101] !== t24 ? (t25 = /* @__PURE__ */ jsx(TreeContext.Provider, {
		value: contextValue,
		children: t24
	}), $[100] = contextValue, $[101] = t24, $[102] = t25) : t25 = $[102];
	let t26;
	return $[103] !== expanded || $[104] !== handleClick || $[105] !== handleKeyDown || $[106] !== id || $[107] !== itemKey || $[108] !== restProps || $[109] !== t22 || $[110] !== t23 || $[111] !== t25 || $[112] !== tabIndex ? (t26 = /* @__PURE__ */ jsxs(StyledTreeItem, {
		"data-selected": t22,
		"data-ui": "TreeItem",
		"data-tree-id": id,
		"data-tree-key": itemKey,
		...restProps,
		"aria-expanded": expanded,
		onClick: handleClick,
		onKeyDown: handleKeyDown,
		ref: setRootElement,
		role: "treeitem",
		tabIndex,
		children: [t23, t25]
	}), $[103] = expanded, $[104] = handleClick, $[105] = handleKeyDown, $[106] = id, $[107] = itemKey, $[108] = restProps, $[109] = t22, $[110] = t23, $[111] = t25, $[112] = tabIndex, $[113] = t26) : t26 = $[113], t26;
}
TreeItem.displayName = "TreeItem";
export { Arrow, Autocomplete, Avatar, AvatarCounter, AvatarStack, Badge, BoundaryElementProvider, Box, Breadcrumbs, Button, Card, Checkbox, Code, CodeSkeleton, ConditionalWrapper, Container, Dialog, DialogContext, DialogProvider, ElementQuery, ErrorBoundary, Flex, Grid, Heading, HeadingSkeleton, Hotkeys, Inline, KBD, Label, LabelSkeleton, Layer, LayerProvider, Menu, MenuButton, MenuDivider, MenuGroup, MenuItem, Popover, Portal, PortalProvider, Radio, Select, Skeleton, Spinner, SrOnly, Stack, Switch, Tab, TabList, TabPanel, Text, TextArea, TextInput, TextSkeleton, ThemeColorProvider, ThemeProvider, Toast, ToastProvider, Tooltip, TooltipDelayGroupContext, TooltipDelayGroupProvider, Tree, TreeItem, VirtualList, _ResizeObserver, _elementSizeObserver, _fillCSSObject, _getArrayProp, _getResponsiveSpace, _hasFocus, _isEnterToClickElement, _isScrollable, _raf, _raf2, _responsive, attemptFocus, containsOrEqualsElement, createColorTheme, focusFirstDescendant, focusLastDescendant, hexToRgb, hslToRgb, isFocusable, isHTMLAnchorElement, isHTMLButtonElement, isHTMLElement, isHTMLInputElement, isHTMLSelectElement, isHTMLTextAreaElement, multiply, parseColor, rem, responsiveCodeFontStyle, responsiveHeadingFont, responsiveLabelFont, responsiveTextAlignStyle, responsiveTextFont, rgbToHex, rgbToHsl, rgba, screen, studioTheme, useArrayProp, useBoundaryElement, useClickOutside, useClickOutsideEvent, useCustomValidity, useDialog, useElementRect, useElementSize, useForwardedRef, useGlobalKeyDown, useLayer, useMatchMedia, useMediaIndex, usePortal, usePrefersDark, usePrefersReducedMotion, useRootTheme, useTheme, useTheme_v2, useToast, useTooltipDelayGroup, useTree };
